<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="../img/favicon.ico">
        <title>BMI - Omics-analysis</title>
        <link href="../css/bootstrap.min.css" rel="stylesheet">
        <link href="../css/font-awesome.min.css" rel="stylesheet">
        <link href="../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css">
        <link href="../stylesheets/extra.css" rel="stylesheet">
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/Bash.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/HTML.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/Java.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/JavaScript.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/Perl.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/Python.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/R.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/rust.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/yaml.min.js"></script>
        <script>hljs.highlightAll();</script> 
    </head>

    <body>
        <div class="navbar fixed-top navbar-expand-lg navbar-dark bg-dark">
            <div class="container">
                <a class="navbar-brand" href="..">Omics-analysis</a>
                <!-- Expander button -->
                <button type="button" class="navbar-toggler" data-toggle="collapse" data-target="#navbar-collapse">
                    <span class="navbar-toggler-icon"></span>
                </button>

                <!-- Expanded navigation -->
                <div id="navbar-collapse" class="navbar-collapse collapse">
                        <!-- Main navigation -->
                        <ul class="nav navbar-nav">
                            <li class="navitem">
                                <a href=".." class="nav-link"></a>
                            </li>
                            <li class="navitem">
                                <a href="../resources/" class="nav-link">Resources</a>
                            </li>
                            <li class="dropdown active">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">Cases <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li>
    <a href="../AF/" class="dropdown-item">Atrial Fibrillation</a>
</li>
                                    
<li>
    <a href="./" class="dropdown-item active">BMI</a>
</li>
                                    
<li>
    <a href="../CAD/" class="dropdown-item">CAD/MI</a>
</li>
                                    
<li>
    <a href="../T2D/" class="dropdown-item">T2D</a>
</li>
                                </ul>
                            </li>
                            <li class="navitem">
                                <a href="../genomics/" class="nav-link">Genomics</a>
                            </li>
                            <li class="navitem">
                                <a href="../proteomics/" class="nav-link">Proteomics</a>
                            </li>
                        </ul>

                    <ul class="nav navbar-nav ml-auto">
                        <li class="nav-item">
                            <a href="#" class="nav-link" data-toggle="modal" data-target="#mkdocs_search_modal">
                                <i class="fa fa-search"></i> Search
                            </a>
                        </li>
                            <li class="nav-item">
                                <a rel="prev" href="../AF/" class="nav-link">
                                    <i class="fa fa-arrow-left"></i> Previous
                                </a>
                            </li>
                            <li class="nav-item">
                                <a rel="next" href="../CAD/" class="nav-link">
                                    Next <i class="fa fa-arrow-right"></i>
                                </a>
                            </li>
                            <li class="nav-item">
                                <a href="https://github.com/jinghuazhao/Omics-analysis" class="nav-link"><i class="fa fa-github"></i> GitHub</a>
                            </li>
                    </ul>
                </div>
            </div>
        </div>

        <div class="container">
            <div class="row">
                    <div class="col-md-3"><div class="navbar-light navbar-expand-md bs-sidebar hidden-print affix" role="complementary">
    <div class="navbar-header">
        <button type="button" class="navbar-toggler collapsed" data-toggle="collapse" data-target="#toc-collapse" title="Table of Contents">
            <span class="fa fa-angle-down"></span>
        </button>
    </div>

    
    <div id="toc-collapse" class="navbar-collapse collapse card bg-secondary">
        <ul class="nav flex-column">
            
            <li class="nav-item" data-level="1"><a href="#bmi" class="nav-link">BMI</a>
              <ul class="nav flex-column">
            <li class="nav-item" data-level="2"><a href="#visualisation" class="nav-link">Visualisation</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#independent-signals" class="nav-link">Independent signals</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#annotation" class="nav-link">Annotation</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#pathway-analysis" class="nav-link">Pathway analysis</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#partitioned-heritabilty" class="nav-link">Partitioned heritabilty</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#mendelian-randomisation" class="nav-link">Mendelian Randomisation</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#twas" class="nav-link">TWAS</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#references" class="nav-link">References</a>
              <ul class="nav flex-column">
              </ul>
            </li>
              </ul>
            </li>
        </ul>
    </div>
</div></div>
                    <div class="col-md-9" role="main">

<h1 id="bmi">BMI</h1>
<p><img src="BMI.png" width="860" height="600" align="right"></p>
<p>We work on the GIANT+Biiobank data on BMI (Yengo et al. 2018), including both the genomewide
<div class="highlight"><pre><span></span><code>wget<span class="w"> </span>https://portals.broadinstitute.org/collaboration/giant/images/0/0f/Meta-analysis_Locke_et_al+UKBiobank_2018.txt.gz
</code></pre></div>
and GCTA --cojo results as used in Mendelian Randomisation analysis downloaded below on the fly. We would refer to <a href="https://github.com/jinghuazhao/software-notes">software-notes</a> where information specific software can be seen from their respective directories.</p>
<h2 id="visualisation">Visualisation</h2>
<p>The first thing is to see the Manhattan plot; in this case the P values could be extremely small and we resort to a
truncated version.
<div class="highlight"><pre><span></span><code><span class="nf">options</span><span class="p">(</span><span class="n">width</span><span class="o">=</span><span class="m">120</span><span class="p">)</span>
<span class="n">gz</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">gzfile</span><span class="p">(</span><span class="s">&quot;Meta-analysis_Locke_et_al+UKBiobank_2018_UPDATED.txt.gz&quot;</span><span class="p">)</span>
<span class="n">BMI</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">within</span><span class="p">(</span><span class="nf">read.delim</span><span class="p">(</span><span class="n">gz</span><span class="p">,</span><span class="n">as.is</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">),</span><span class="w"> </span><span class="p">{</span><span class="n">Z</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">BETA</span><span class="o">/</span><span class="n">SE</span><span class="p">})</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">subset</span><span class="p">(</span><span class="n">BMI</span><span class="p">[</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;CHR&quot;</span><span class="p">,</span><span class="s">&quot;POS&quot;</span><span class="p">,</span><span class="s">&quot;SNP&quot;</span><span class="p">,</span><span class="s">&quot;P&quot;</span><span class="p">)],</span><span class="n">CHR</span><span class="o">!=</span><span class="m">16</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">P</span><span class="o">&lt;=</span><span class="m">1e-150</span><span class="p">))</span>
<span class="nf">library</span><span class="p">(</span><span class="n">Rmpfr</span><span class="p">)</span>
<span class="nf">print</span><span class="p">(</span><span class="nf">within</span><span class="p">(</span><span class="nf">subset</span><span class="p">(</span><span class="n">BMI</span><span class="p">,</span><span class="w"> </span><span class="n">P</span><span class="o">==</span><span class="m">0</span><span class="p">,</span><span class="w"> </span><span class="n">select</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="n">CHR</span><span class="p">,</span><span class="n">POS</span><span class="p">,</span><span class="n">SNP</span><span class="p">,</span><span class="n">Z</span><span class="p">)),</span><span class="w"> </span>
<span class="w">             </span><span class="p">{</span><span class="n">P</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">format</span><span class="p">(</span><span class="m">2</span><span class="o">*</span><span class="nf">pnorm</span><span class="p">(</span><span class="nf">mpfr</span><span class="p">(</span><span class="nf">abs</span><span class="p">(</span><span class="n">Z</span><span class="p">),</span><span class="m">100</span><span class="p">),</span><span class="n">lower.tail</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">));</span><span class="w"> </span><span class="n">Pvalue</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">pvalue</span><span class="p">(</span><span class="n">Z</span><span class="p">);</span><span class="w"> </span><span class="n">log10P</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="o">-</span><span class="nf">log10p</span><span class="p">(</span><span class="n">Z</span><span class="p">)}))</span>
<span class="nf">png</span><span class="p">(</span><span class="s">&quot;BMI.png&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">res</span><span class="o">=</span><span class="m">300</span><span class="p">,</span><span class="w"> </span><span class="n">units</span><span class="o">=</span><span class="s">&quot;in&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">width</span><span class="o">=</span><span class="m">9</span><span class="p">,</span><span class="w"> </span><span class="n">height</span><span class="o">=</span><span class="m">6</span><span class="p">)</span>
<span class="nf">par</span><span class="p">(</span><span class="n">oma</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">0</span><span class="p">,</span><span class="m">0</span><span class="p">,</span><span class="m">0</span><span class="p">),</span><span class="w"> </span><span class="n">mar</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="m">5</span><span class="p">,</span><span class="m">6.5</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="m">1</span><span class="p">))</span>
<span class="nf">mhtplot.trunc</span><span class="p">(</span><span class="n">BMI</span><span class="p">,</span><span class="w"> </span><span class="n">chr</span><span class="o">=</span><span class="s">&quot;CHR&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">bp</span><span class="o">=</span><span class="s">&quot;POS&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">z</span><span class="o">=</span><span class="s">&quot;Z&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">snp</span><span class="o">=</span><span class="s">&quot;SNP&quot;</span><span class="p">,</span>
<span class="w">              </span><span class="n">suggestiveline</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">genomewideline</span><span class="o">=-</span><span class="nf">log10</span><span class="p">(</span><span class="m">1e-8</span><span class="p">),</span>
<span class="w">              </span><span class="n">cex.mtext</span><span class="o">=</span><span class="m">1.2</span><span class="p">,</span><span class="w"> </span><span class="n">cex.text</span><span class="o">=</span><span class="m">1.2</span><span class="p">,</span>
<span class="w">              </span><span class="n">annotatelog10P</span><span class="o">=</span><span class="m">156</span><span class="p">,</span><span class="w"> </span><span class="n">annotateTop</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">highlight</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;rs13021737&quot;</span><span class="p">,</span><span class="s">&quot;rs17817449&quot;</span><span class="p">,</span><span class="s">&quot;rs6567160&quot;</span><span class="p">),</span>
<span class="w">              </span><span class="n">mtext.line</span><span class="o">=</span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="n">y.brk1</span><span class="o">=</span><span class="m">200</span><span class="p">,</span><span class="w"> </span><span class="n">y.brk2</span><span class="o">=</span><span class="m">280</span><span class="p">,</span><span class="w"> </span><span class="n">cex.axis</span><span class="o">=</span><span class="m">1.2</span><span class="p">,</span><span class="w"> </span><span class="n">cex.y</span><span class="o">=</span><span class="m">1.2</span><span class="p">,</span><span class="w"> </span><span class="n">cex</span><span class="o">=</span><span class="m">0.5</span><span class="p">,</span>
<span class="w">              </span><span class="n">y.ax.space</span><span class="o">=</span><span class="m">20</span><span class="p">,</span>
<span class="w">              </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;blue4&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;skyblue&quot;</span><span class="p">)</span>
<span class="p">)</span>
<span class="nf">dev.off</span><span class="p">()</span>
</code></pre></div>
Note especially those P values equal to zero -- <code>pvalue()</code> and <code>R/Rmpfr</code> both give the minimum approximately 1.26e-479. Nevertheless, it is safer
to generate -log10(P) on the fly -- in <a href="BMI.png">the plot</a> chromosome.16 stands out which would not be so should we restrict ourselves only
to nonzero P values. We went further to highlight three SNPs.</p>
<h2 id="independent-signals">Independent signals</h2>
<p><div class="highlight"><pre><span></span><code><span class="o">(</span>
<span class="w">  </span>zcat<span class="w"> </span>Meta-analysis_Locke_et_al+UKBiobank_2018_UPDATED.txt.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>head<span class="w"> </span>-1
<span class="w">  </span>zcat<span class="w"> </span>Meta-analysis_Locke_et_al+UKBiobank_2018_UPDATED.txt.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>awk<span class="w"> </span><span class="s1">&#39;(NR&gt;1 &amp;&amp; $9&lt;=1e-8)&#39;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>sort<span class="w"> </span>-k1,1n<span class="w"> </span>-k2,2n
<span class="o">)</span><span class="w"> </span>&gt;<span class="w"> </span>BMI.dat
module<span class="w"> </span>load<span class="w"> </span>gcc/5.2.0
R<span class="w"> </span>--no-save<span class="w"> </span>-q<span class="w"> </span>&lt;<span class="w"> </span>BMI.R<span class="w"> </span>&gt;<span class="w"> </span>BMI.out
</code></pre></div>
where the <code>module</code> command prepares for appropriate version of R to be enabled and <code>BMI.R</code> selects independent signals as follows,
<div class="highlight"><pre><span></span><code><span class="nf">options</span><span class="p">(</span><span class="n">echo</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span>
<span class="n">BMI</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.delim</span><span class="p">(</span><span class="s">&quot;BMI.dat&quot;</span><span class="p">,</span><span class="n">as.is</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)[</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;CHR&quot;</span><span class="p">,</span><span class="s">&quot;POS&quot;</span><span class="p">,</span><span class="s">&quot;SNP&quot;</span><span class="p">,</span><span class="s">&quot;BETA&quot;</span><span class="p">,</span><span class="s">&quot;SE&quot;</span><span class="p">,</span><span class="s">&quot;P&quot;</span><span class="p">)]</span>
<span class="nf">require</span><span class="p">(</span><span class="n">reshape</span><span class="p">)</span>
<span class="n">BMI</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">rename</span><span class="p">(</span><span class="n">BMI</span><span class="p">,</span><span class="nf">c</span><span class="p">(</span><span class="n">CHR</span><span class="o">=</span><span class="s">&quot;Chrom&quot;</span><span class="p">,</span><span class="n">POS</span><span class="o">=</span><span class="s">&quot;End&quot;</span><span class="p">,</span><span class="n">SNP</span><span class="o">=</span><span class="s">&quot;MarkerName&quot;</span><span class="p">,</span><span class="n">BETA</span><span class="o">=</span><span class="s">&quot;Effect&quot;</span><span class="p">,</span><span class="n">SE</span><span class="o">=</span><span class="s">&quot;StdErr&quot;</span><span class="p">,</span><span class="n">P</span><span class="o">=</span><span class="s">&quot;P.value&quot;</span><span class="p">))</span>
<span class="nf">require</span><span class="p">(</span><span class="n">gap</span><span class="p">)</span>
<span class="n">chrs</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">with</span><span class="p">(</span><span class="n">BMI</span><span class="p">,</span><span class="nf">unique</span><span class="p">(</span><span class="n">Chrom</span><span class="p">))</span>
<span class="nf">for</span><span class="p">(</span><span class="n">chr</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="n">chrs</span><span class="p">)</span>
<span class="p">{</span>
<span class="w">  </span><span class="nf">print</span><span class="p">(</span><span class="n">chr</span><span class="p">)</span>
<span class="w">  </span><span class="n">ps</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">subset</span><span class="p">(</span><span class="n">BMI</span><span class="p">[</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;Chrom&quot;</span><span class="p">,</span><span class="s">&quot;End&quot;</span><span class="p">,</span><span class="s">&quot;MarkerName&quot;</span><span class="p">,</span><span class="s">&quot;Effect&quot;</span><span class="p">,</span><span class="s">&quot;StdErr&quot;</span><span class="p">)],</span><span class="n">Chrom</span><span class="o">==</span><span class="n">chr</span><span class="p">)</span>
<span class="w">  </span><span class="nf">row.names</span><span class="p">(</span><span class="n">ps</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="nf">nrow</span><span class="p">(</span><span class="n">ps</span><span class="p">)</span>
<span class="w">  </span><span class="nf">sentinels</span><span class="p">(</span><span class="n">ps</span><span class="p">,</span><span class="n">chr</span><span class="p">,</span><span class="m">1</span><span class="p">)</span>
<span class="p">}</span>
</code></pre></div>
It is noted that awk has problem dealing with very small values and the code is altered slightly as follows,
<div class="highlight"><pre><span></span><code><span class="o">(</span>
<span class="w">  </span>zcat<span class="w"> </span>Meta-analysis_Locke_et_al+UKBiobank_2018_UPDATED.txt.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>head<span class="w"> </span>-1
<span class="w">  </span>zcat<span class="w"> </span>Meta-analysis_Locke_et_al+UKBiobank_2018_UPDATED.txt.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>awk<span class="w"> </span><span class="s1">&#39;</span>
<span class="s1">    function abs(x)</span>
<span class="s1">    {</span>
<span class="s1">      if (x&lt;0) return -x;</span>
<span class="s1">      else return x;</span>
<span class="s1">    }</span>
<span class="s1">    (NR&gt;1 &amp;&amp; abs($7/$8) &gt;= 5.730729)&#39;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="se">\</span>
<span class="w">  </span>sort<span class="w"> </span>-k1,1n<span class="w"> </span>-k2,2n
<span class="o">)</span><span class="w"> </span>&gt;<span class="w"> </span>BMI.dat

R<span class="w"> </span>--no-save<span class="w"> </span>-q<span class="w"> </span>&lt;<span class="w"> </span>BMI.R<span class="w"> </span>&gt;<span class="w"> </span>BMI.out
</code></pre></div>
Then BMI.out has 632 entries instead of 626, i.e., six were recovered. Note also the magic value 5.730729 is obtained from <code>qnorm(1-0.5e-8)</code> from R.</p>
<h2 id="annotation">Annotation</h2>
<p>As a follow-up to the earlier GIANT analysis, we have in the directory results from PhenoScanner for the 97 SNPs described in the SUMSTATS repository
<div class="highlight"><pre><span></span><code>module<span class="w"> </span>load<span class="w"> </span>phenoscanner/phenoscanner_v2
phenoscanner<span class="w"> </span>-c<span class="w"> </span>All<span class="w"> </span>-l<span class="w"> </span>No<span class="w"> </span>-p<span class="w"> </span><span class="m">0</span>.00001<span class="w"> </span>-i<span class="w"> </span><span class="m">97</span>.snps<span class="w"> </span>-o<span class="w"> </span><span class="m">97</span>
</code></pre></div>
For the GIANT+UKB data above, the setup is
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;NR&gt;1&#39;</span><span class="w"> </span>BMI.dat<span class="w"> </span><span class="p">|</span><span class="w"> </span><span class="se">\</span>
cut<span class="w"> </span>-f3<span class="w"> </span>&gt;<span class="w"> </span>BMI.rsid
phenoscanner<span class="w"> </span>-c<span class="w"> </span>All<span class="w"> </span>-l<span class="w"> </span>No<span class="w"> </span>-p<span class="w"> </span><span class="m">0</span>.00001<span class="w"> </span>-i<span class="w"> </span>BMI.rsid<span class="w"> </span>-o<span class="w"> </span>BMI
</code></pre></div>
in both cases, results could be expanded by allowing for LD.</p>
<h2 id="pathway-analysis">Pathway analysis</h2>
<p><div class="highlight"><pre><span></span><code>gunzip<span class="w"> </span>-c<span class="w"> </span>Meta-analysis_Locke_et_al+UKBiobank_2018.txt.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span>awk<span class="w"> </span><span class="s1">&#39;</span>
<span class="s1">{</span>
<span class="s1">   FS=OFS=&quot;\t&quot;</span>
<span class="s1">   if(NR==1) print &quot;SNP&quot;,&quot;Chr&quot;,&quot;Pos&quot;,&quot;P&quot;</span>
<span class="s1">   else if($9&lt;=5e-8) print $3,$1,$2,$9</span>
<span class="s1">}&#39;</span><span class="w"> </span><span class="p">|</span><span class="w"> </span>gzip<span class="w"> </span>-f<span class="w"> </span>&gt;<span class="w"> </span>BMI.txt.gz
</code></pre></div>
where we opt to customise the header/column rather than the DEPICT configuration file. Moreover, the (hg19) chromosomal positions are <em>eventually</em> back in the data which would facilitate GCTA --cojo analysis and mirrors https://github.com/jinghuazhao/SUMSTATS.</p>
<p>As usual, we make a call to <a href="BMI.cfg">BMI.cfg</a> via
<div class="highlight"><pre><span></span><code>depict.py<span class="w"> </span>BMI.cfg
</code></pre></div>
where the <code>DEPICT_v1_rel194.tar.gz</code> version is used. Once started, there was complaint that
<div class="highlight"><pre><span></span><code>Retrieving background loci
Exiting.. To few background files in data/backgrounds/nloci723_nperm500_kb500_rsq0.1_mhc25000000-35000000_colld0.5-collection-1000genomespilot-depict-150429/. Please remove the folder, rerun DEPICT and contact tunepers@broadinstitute.org if the error prevails.
</code></pre></div>
Follow instruction and remove the directory. It is very slow-going, ~20 hours on our Linux node but surprisingly half that time under my Windows 10 whose directory zipped and then unzipped under Linux and run<code>depict.py</code>there.</p>
<p>We then generate <a href="BMI.xlsx">BMI.xlsx</a> as in <a href="https://github.com/jinghuazhao/PW-pipeline/wiki">PW-pipelne</a>. While there are 859 genesets with FDR&lt;0.05, tissue enrichment shows compelingly an overwhelming role of the nervous system.</p>
<p>This is detailed from <a href="https://github.com/jinghuazhao/PW-pipeline/wiki">PW-pipeline/wiki</a> for the analysis.</p>
<h2 id="partitioned-heritabilty">Partitioned heritabilty</h2>
<p>Information for the documentation example is available from <a href="https://github.com/jinghuazhao/software-notes/">software-notes</a>. Here we carry on with the .gz file above.
<div class="highlight"><pre><span></span><code>gunzip<span class="w"> </span>-c<span class="w"> </span>Meta-analysis_Locke_et_al+UKBiobank_2018.txt.gz<span class="w"> </span>&gt;<span class="w"> </span>ldsc.txt
python<span class="w"> </span>munge_sumstats.py<span class="w"> </span>--sumstats<span class="w"> </span>ldsc.txt<span class="w"> </span>--a1<span class="w"> </span>Tested_Allele<span class="w"> </span>--a2<span class="w"> </span>Other_allele<span class="w"> </span>--merge-alleles<span class="w"> </span>w_hm3.snplist<span class="w"> </span>--out<span class="w"> </span>ldsc<span class="w"> </span>--a1-inc
</code></pre></div>
where we fix the header with appropriate command-line parameters. We see <a href="ldsc.sumstats.gz">ldsc.sumstats.gz</a> and <a href="ldsc.log">ldsc.log</a> and carry on with
<div class="highlight"><pre><span></span><code>python<span class="w"> </span>ldsc.py<span class="w"> </span>--h2<span class="w"> </span>ldsc.sumstats.gz<span class="se">\</span>
<span class="w">        </span>--ref-ld-chr<span class="w"> </span>baseline_v1.1/baseline.<span class="se">\</span>
<span class="w">        </span>--w-ld-chr<span class="w"> </span>1000G_Phase3_weights_hm3_no_MHC/weights.hm3_noMHC.<span class="se">\</span>
<span class="w">        </span>--overlap-annot<span class="se">\</span>
<span class="w">        </span>--frqfile-chr<span class="w"> </span>1000G_Phase3_frq/1000G.EUR.QC.<span class="se">\</span>
<span class="w">        </span>--out<span class="w"> </span>ldsc_baseline
</code></pre></div>
to generate <a href="ldsc_baseline.results">ldsc_baseline.results</a> and <a href="ldsc_baseline.log">ldsc_baseline.log</a> and
<div class="highlight"><pre><span></span><code>python<span class="w"> </span>ldsc.py<span class="w"> </span>--h2<span class="w"> </span>ldsc.sumstats.gz<span class="se">\</span>
<span class="w">        </span>--w-ld-chr<span class="w"> </span>1000G_Phase3_weights_hm3_no_MHC/weights.hm3_noMHC.<span class="se">\</span>
<span class="w">        </span>--ref-ld-chr<span class="w"> </span>1000G_Phase3_cell_type_groups/cell_type_group.3.,baseline_v1.1/baseline.<span class="se">\</span>
<span class="w">        </span>--overlap-annot<span class="se">\</span>
<span class="w">        </span>--frqfile-chr<span class="w"> </span>1000G_Phase3_frq/1000G.EUR.QC.<span class="se">\</span>
<span class="w">        </span>--out<span class="w"> </span>ldsc_CNS<span class="se">\</span>
<span class="w">        </span>--print-coefficients
</code></pre></div>
for <a href="ldsc_CNS.results">ldsc_baseline.results</a> and <a href="ldsc_CNS.log">ldsc_baseline.log</a>.</p>
<h2 id="mendelian-randomisation">Mendelian Randomisation</h2>
<p>A documented example on BMI-lung cancer is adapted in <a href="https://github.com/jinghuazhao/software-notes">software-notes</a> but our focus here is on BMI-T2D from DIAGRAM,
<div class="highlight"><pre><span></span><code>wget<span class="w"> </span>-qO-<span class="w"> </span>https://portals.broadinstitute.org/collaboration/giant/images/e/e2/Meta-analysis_Locke_et_al+UKBiobank_2018_top_941_from_COJO_analysis_UPDATED.txt.gz<span class="w"> </span>&gt;<span class="w"> </span>BMI-COJO.gz
R<span class="w"> </span>--no-save<span class="w"> </span>-q<span class="w"> </span>&lt;<span class="w"> </span>MR.R<span class="w"> </span>&gt;<span class="w"> </span>MR.log
</code></pre></div>
and call <a href="MR.R">MR.R</a> to generate <a href="MR.log">MR.log</a> and <a href="MR.pdf">MR.pdf</a>.</p>
<h2 id="twas">TWAS</h2>
<p>We start with <code>MetaXcan</code> as follows,
<div class="highlight"><pre><span></span><code><span class="nb">cd</span><span class="w"> </span>/home/jhz22/D/genetics/hakyimlab/MetaXcan/software

./MetaXcan.py<span class="w"> </span><span class="se">\</span>
--model_db_path<span class="w"> </span>/home/jhz22/D/genetics/hakyimlab/PredictDB/GTEx-V7_HapMap-2017-11-29/gtex_v7_Brain_Amygdala_imputed_europeans_tw_0.5_signif.db<span class="w"> </span><span class="se">\</span>
--covariance<span class="w"> </span>/home/jhz22/D/genetics/hakyimlab/PredictDB/GTEx-V7_HapMap-2017-11-29/gtex_v7_Brain_Amygdala_imputed_eur_covariances.txt.gz<span class="w"> </span><span class="se">\</span>
--gwas_file<span class="w"> </span>ldsc.txt<span class="w"> </span><span class="se">\</span>
--snp_column<span class="w"> </span>SNP<span class="w"> </span><span class="se">\</span>
--effect_allele_column<span class="w"> </span>Tested_Allele<span class="w"> </span><span class="se">\</span>
--non_effect_allele_column<span class="w"> </span>Other_Allele<span class="w"> </span><span class="se">\</span>
--beta_column<span class="w"> </span>BETA<span class="w"> </span><span class="se">\</span>
--pvalue_column<span class="w"> </span>P<span class="w"> </span><span class="se">\</span>
--output_file<span class="w"> </span>MX.csv
</code></pre></div>
where instead of chromosome-specific summary statistics as shown in <a href="https://github.com/jinghuazhao/software-notes">software-notes</a> we use <code>ldsc.txt</code> created above directly with results contained in <a href="MX.csv">MX.csv</a> and screen output <a href="MX.log">MX.log</a>.</p>
<p>We would also be tempting to contrast results with FUSION,
<div class="highlight"><pre><span></span><code>gunzip<span class="w"> </span>-c<span class="w"> </span>Meta-analysis_Locke_et_al+UKBiobank_2018.txt.gz<span class="w"> </span><span class="p">|</span><span class="w"> </span>awk<span class="w"> </span><span class="s1">&#39;</span>
<span class="s1">{</span>
<span class="s1">   FS=OFS=&quot;\t&quot;</span>
<span class="s1">   if(NR==1) print &quot;SNP&quot;,&quot;A1&quot;,&quot;A2&quot;,&quot;Z&quot;</span>
<span class="s1">   else print $3,$4,$5,$7/$8</span>
<span class="s1">}&#39;</span><span class="w"> </span>&gt;<span class="w"> </span>fusion.txt
<span class="k">for</span><span class="w"> </span>chr<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="k">$(</span>seq<span class="w"> </span><span class="m">22</span><span class="k">)</span>
<span class="k">do</span>
<span class="w">    </span>Rscript<span class="w"> </span>FUSION.assoc_test.R<span class="w"> </span>--sumstats<span class="w"> </span>fusion.txt<span class="w"> </span><span class="se">\</span>
<span class="w">                                </span>--weight1s<span class="w"> </span>WEIGHTS/NTR.BLOOD.RNAARR.pos<span class="w"> </span>--weights_dir<span class="w"> </span>WEIGHTS/<span class="w"> </span><span class="se">\</span>
<span class="w">                                </span>--ref_ld_chr<span class="w"> </span>LDREF/1000G.EUR.<span class="w"> </span>--chr<span class="w"> </span><span class="nv">$chr</span><span class="w"> </span>--out<span class="w"> </span>fusion.<span class="nv">$chr</span>.dat
<span class="k">done</span>
</code></pre></div>
Note again the header/column is customised differently from its DEPICT counterpart and results are obtained by chromosome.</p>
<p>Two additional aspects are useful to explore:</p>
<h3 id="fine-mapping">Fine-mapping</h3>
<p>In line with the fact that both TWAS and GWAS z scores are available, the option <code>--caviar</code> natually put them as input files for <code>CARIAR</code>,
<div class="highlight"><pre><span></span><code><span class="k">for</span><span class="w"> </span>chr<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="k">$(</span>seq<span class="w"> </span><span class="m">22</span><span class="k">)</span>
<span class="k">do</span>
Rscript<span class="w"> </span>FUSION.assoc_test.R<span class="w"> </span><span class="se">\</span>
--sumstats<span class="w"> </span>fusion.txt<span class="w"> </span><span class="se">\</span>
--weights<span class="w"> </span>/home/jhz22/D/mrc/genetics/FUSION/GE/CMC.BRAIN.RNASEQ.pos<span class="w"> </span><span class="se">\</span>
--weights_dir<span class="w"> </span>/home/jhz22/D/mrc/genetics/FUSION/GE/<span class="w"> </span><span class="se">\</span>
--ref_ld_chr<span class="w"> </span>/home/jhz22/D/genetics/fusion_twas/LDREF/1000G.EUR.<span class="w"> </span><span class="se">\</span>
--chr<span class="w"> </span><span class="nv">$chr</span><span class="w"> </span><span class="se">\</span>
--caviar<span class="w"> </span><span class="se">\</span>
--out<span class="w"> </span>caviar
<span class="k">done</span>
</code></pre></div>
which are CAVIAR.EQTL.Z, CAVIAR.GWAS.Z, CAVIAR.LD triplets with prefix caviar.genename.</p>
<h3 id="colocalisation">Colocalisation</h3>
<p>This is furnished with <code>--coloc_P --GWASN</code>,
<div class="highlight"><pre><span></span><code><span class="k">for</span><span class="w"> </span>chr<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="k">$(</span>seq<span class="w"> </span><span class="m">22</span><span class="k">)</span>
<span class="k">do</span>
Rscript<span class="w"> </span>FUSION.assoc_test.R<span class="w"> </span><span class="se">\</span>
--sumstats<span class="w"> </span>fusion.txt<span class="w"> </span><span class="se">\</span>
--weights<span class="w"> </span>/home/jhz22/D/mrc/genetics/FUSION/GE/CMC.BRAIN.RNASEQ.pos<span class="w"> </span><span class="se">\</span>
--weights_dir<span class="w"> </span>/home/jhz22/D/mrc/genetics/FUSION/GE/<span class="w"> </span><span class="se">\</span>
--ref_ld_chr<span class="w"> </span>/home/jhz22/D/genetics/fusion_twas/LDREF/1000G.EUR.<span class="w"> </span><span class="se">\</span>
--chr<span class="w"> </span><span class="nv">$chr</span><span class="w"> </span><span class="se">\</span>
--coloc_P<span class="w"> </span>5e-8<span class="w"> </span><span class="se">\</span>
--GWASN<span class="w"> </span><span class="m">70000</span><span class="w"> </span><span class="se">\</span>
--out<span class="w"> </span>coloc<span class="nv">$chr</span>.dat
<span class="k">done</span>
</code></pre></div></p>
<h3 id="prediction">Prediction</h3>
<p>See <a href="https://github.com/jinghuazhao/software-notes">software-notes</a> on the use of <code>utils/make_score.R</code>, which is based on the best model.</p>
<h3 id="approximately-independent-ld-blocks">Approximately independent LD blocks</h3>
<p>We can use these blocks genomewide or a specific gene, e.g., MC4R,
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span><span class="s1">&#39;NR&gt;1&#39;</span><span class="w"> </span>ldsc.txt<span class="w"> </span><span class="p">|</span><span class="w"> </span>sort<span class="w"> </span>-k1,1n<span class="w"> </span>-k2,2n<span class="w"> </span><span class="p">|</span><span class="w"> </span>awk<span class="w"> </span><span class="s1">&#39;</span>
<span class="s1">{</span>
<span class="s1">  OFS=&quot;\t&quot;</span>
<span class="s1">  if (NR==1) print &quot;#chrom&quot;, &quot;Start&quot;, &quot;End&quot;, &quot;SNP&quot;, &quot;A1&quot;, &quot;A2&quot;, &quot;FreqA1&quot;, &quot;BETA&quot;, &quot;SE&quot;, &quot;P&quot;, &quot;N&quot;</span>
<span class="s1">  print &quot;chr&quot; $1, $2-1, $2, $3, $4, $5, $6, $7, $8, $9, $10</span>
<span class="s1">}&#39;</span><span class="w"> </span>&gt;<span class="w"> </span>BMI.bed

<span class="c1"># EUR.bed is now with FM-pipeline</span>
intersectBed<span class="w"> </span>-a<span class="w"> </span>BMI.bed<span class="w"> </span>-b<span class="w"> </span>EUR.bed<span class="w"> </span>-loj<span class="w"> </span><span class="p">|</span><span class="w"> </span>cut<span class="w"> </span>-f12-14<span class="w"> </span>--complement<span class="w"> </span>&gt;<span class="w"> </span>BMI.txt

cut<span class="w"> </span>-f4,10<span class="w"> </span>BMI.txt<span class="w"> </span>&gt;<span class="w"> </span>BMI.snpandp
<span class="c1"># gene-based association</span>
vegas2v2<span class="w"> </span>-G<span class="w"> </span>-snpandp<span class="w"> </span>BMI.snpandp<span class="w"> </span>-custom<span class="w"> </span><span class="nv">$PWD</span>/g1000p3_EUR<span class="w"> </span>-glist<span class="w"> </span>glist-hg19<span class="w"> </span>-out<span class="w"> </span>genes
<span class="c1"># pathway-based association</span>
awk<span class="w"> </span><span class="s1">&#39;(NR&gt;1){OFS=&quot;\t&quot;;gsub(/&quot;/,&quot;&quot;,$0);print $2,$8}&#39;</span><span class="w"> </span>genes.out<span class="w"> </span>&gt;<span class="w"> </span>BMI.geneandp
vegas2v2<span class="w"> </span>-P<span class="w"> </span>-geneandp<span class="w"> </span>BMI.geneandp<span class="w"> </span>-glist<span class="w"> </span>glist-hg19<span class="w"> </span>-geneandpath<span class="w"> </span>biosystems20160324.vegas2pathSYM<span class="w"> </span>-out<span class="w"> </span>pathways

<span class="c1"># grep MC4R glist-hg19 --&gt; 18 58038563 58040001 MC4R</span>
<span class="c1"># 1439 bp</span>
<span class="c1"># EUR.bed -- &gt; 1563   chr18 57630483 59020751 region1563</span>
<span class="c1"># 1390268 bp</span>

awk<span class="w"> </span>/region1563/<span class="w"> </span>BMI.txt<span class="w"> </span>&gt;<span class="w"> </span>MC4R.txt

<span class="c1"># wc -l MC4R.txt</span>
<span class="c1"># 1342 SNPs</span>

cut<span class="w"> </span>-f4,10<span class="w"> </span>MC4R.txt<span class="w"> </span>&gt;<span class="w"> </span>MC4R.snpandp
<span class="nb">echo</span><span class="w"> </span>MC4R<span class="w"> </span>&gt;<span class="w"> </span>MC4R.genelist
vegas2v2<span class="w"> </span>-G<span class="w"> </span>-snpandp<span class="w"> </span>MC4R.snpandp<span class="w"> </span>-custom<span class="w"> </span><span class="nv">$PWD</span>/g1000p3_EUR<span class="w"> </span>-glist<span class="w"> </span>glist-hg19<span class="w"> </span>-genelist<span class="w"> </span>MC4R.genelist<span class="w"> </span>-out<span class="w"> </span>MC4R
</code></pre></div>
The reason to use ldsc.txt is to allow for the possibility of finemapping.</p>
<h3 id="colocalisation-analysis">Colocalisation analysis</h3>
<p>This can be explored with https://github.com/joepickrell/gwas-pw. Assuming that bmi_height.gz has per-SNP z values and variances, the analysis proceeds with
```bash
gwas-pw -i bmi_height.gz -bed EUR.bed -phenos BMI HEIGHT
````
giving results on four models (1. BMI only, 2. HEIGHT only, 3. both, 4. separate) significance.</p>
<h2 id="references">References</h2>
<p>Yengo L, et al (2018). Meta-analysis of genome-wide association studies for height and body mass index in ~700,000 individuals of European ancestry. <em>Hum Mol Genet</em> 27(20):3641â€“3649, BioRxiv,
https://www.biorxiv.org/content/10.1101/274654v2.</p></div>
            </div>
        </div>

        <footer class="col-md-12">
            <hr>
            <p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>
        <script src="../js/jquery-3.6.0.min.js"></script>
        <script src="../js/bootstrap.min.js"></script>
        <script>
            var base_url = "..",
                shortcuts = {"help": 191, "next": 78, "previous": 80, "search": 83};
        </script>
        <script src="../js/base.js"></script>
        <script src="../javascripts/config.js"></script>
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
        <script src="../search/main.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="searchModalLabel" aria-hidden="true">
    <div class="modal-dialog modal-lg">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="searchModalLabel">Search</h4>
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            </div>
            <div class="modal-body">
                <p>From here you can search these documents. Enter your search terms below.</p>
                <form>
                    <div class="form-group">
                        <input type="search" class="form-control" placeholder="Search..." id="mkdocs-search-query" title="Type search term here">
                    </div>
                </form>
                <div id="mkdocs-search-results" data-no-results-text="No results found"></div>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div><div class="modal" id="mkdocs_keyboard_modal" tabindex="-1" role="dialog" aria-labelledby="keyboardModalLabel" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="keyboardModalLabel">Keyboard Shortcuts</h4>
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            </div>
            <div class="modal-body">
              <table class="table">
                <thead>
                  <tr>
                    <th style="width: 20%;">Keys</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td class="help shortcut"><kbd>?</kbd></td>
                    <td>Open this help</td>
                  </tr>
                  <tr>
                    <td class="next shortcut"><kbd>n</kbd></td>
                    <td>Next page</td>
                  </tr>
                  <tr>
                    <td class="prev shortcut"><kbd>p</kbd></td>
                    <td>Previous page</td>
                  </tr>
                  <tr>
                    <td class="search shortcut"><kbd>s</kbd></td>
                    <td>Search</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div>

    </body>
</html>
