<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="../img/favicon.ico">
        <title>Resources - Omics-analysis</title>
        <link href="../css/bootstrap.min.css" rel="stylesheet">
        <link href="../css/font-awesome.min.css" rel="stylesheet">
        <link href="../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css">
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/Bash.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/HTML.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/Java.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/JavaScript.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/Perl.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/Python.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/R.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/rust.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/yaml.min.js"></script>
        <script>hljs.highlightAll();</script> 
    </head>

    <body>
        <div class="navbar fixed-top navbar-expand-lg navbar-dark bg-dark">
            <div class="container">
                <a class="navbar-brand" href="..">Omics-analysis</a>
                <!-- Expander button -->
                <button type="button" class="navbar-toggler" data-toggle="collapse" data-target="#navbar-collapse">
                    <span class="navbar-toggler-icon"></span>
                </button>

                <!-- Expanded navigation -->
                <div id="navbar-collapse" class="navbar-collapse collapse">
                        <!-- Main navigation -->
                        <ul class="nav navbar-nav">
                            <li class="navitem">
                                <a href=".." class="nav-link"></a>
                            </li>
                            <li class="navitem active">
                                <a href="./" class="nav-link">Resources</a>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">Cases <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li>
    <a href="../AF/" class="dropdown-item">Atrial Fibrillation</a>
</li>
                                    
<li>
    <a href="../BMI/" class="dropdown-item">BMI</a>
</li>
                                    
<li>
    <a href="../CAD/" class="dropdown-item">CAD/MI</a>
</li>
                                    
<li>
    <a href="../T2D/" class="dropdown-item">T2D</a>
</li>
                                </ul>
                            </li>
                            <li class="navitem">
                                <a href="../genomics/" class="nav-link">Genomics</a>
                            </li>
                            <li class="navitem">
                                <a href="../proteomics/" class="nav-link">Proteomics</a>
                            </li>
                        </ul>

                    <ul class="nav navbar-nav ml-auto">
                        <li class="nav-item">
                            <a href="#" class="nav-link" data-toggle="modal" data-target="#mkdocs_search_modal">
                                <i class="fa fa-search"></i> Search
                            </a>
                        </li>
                            <li class="nav-item">
                                <a rel="prev" href=".." class="nav-link">
                                    <i class="fa fa-arrow-left"></i> Previous
                                </a>
                            </li>
                            <li class="nav-item">
                                <a rel="next" href="../AF/" class="nav-link">
                                    Next <i class="fa fa-arrow-right"></i>
                                </a>
                            </li>
                            <li class="nav-item">
                                <a href="https://github.com/jinghuazhao/Omics-analysis" class="nav-link"><i class="fa fa-github"></i> GitHub</a>
                            </li>
                    </ul>
                </div>
            </div>
        </div>

        <div class="container">
            <div class="row">
                    <div class="col-md-3"><div class="navbar-light navbar-expand-md bs-sidebar hidden-print affix" role="complementary">
    <div class="navbar-header">
        <button type="button" class="navbar-toggler collapsed" data-toggle="collapse" data-target="#toc-collapse" title="Table of Contents">
            <span class="fa fa-angle-down"></span>
        </button>
    </div>

    
    <div id="toc-collapse" class="navbar-collapse collapse card bg-secondary">
        <ul class="nav flex-column">
            
            <li class="nav-item" data-level="1"><a href="#resources" class="nav-link">Resources</a>
              <ul class="nav flex-column">
            <li class="nav-item" data-level="2"><a href="#annotation" class="nav-link">Annotation</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#biobanks" class="nav-link">Biobanks</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#catalog" class="nav-link">Catalog</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#efo" class="nav-link">EFO</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#eqtlgen" class="nav-link">eQTLGen</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#metamaplite" class="nav-link">MetaMapLite</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#mr-baseopengwas" class="nav-link">MR-Base/OpenGWAS</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#opentargets" class="nav-link">OpenTargets</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#rentrez" class="nav-link">rentrez</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#roadmap" class="nav-link">Roadmap</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#snakemake-workflow-catalogue" class="nav-link">snakemake workflow catalogue</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#twas" class="nav-link">TWAS</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#other-links" class="nav-link">Other links</a>
              <ul class="nav flex-column">
              </ul>
            </li>
              </ul>
            </li>
        </ul>
    </div>
</div></div>
                    <div class="col-md-9" role="main">

<h1 id="resources">Resources</h1>
<h2 id="annotation">Annotation</h2>
<p><a href="https://www.ensembl.org/info/data/mysql.html">The Ensembl public MySQL Servers</a></p>
<p>The following script gives information on genes from ENSEMBL as well as attributes (columns) that contains <code>gene</code>.</p>
<pre><code class="language-r">library(biomaRt)
listMarts()
mart &lt;- useMart(&quot;ENSEMBL_MART_FUNCGEN&quot;)
listDatasets(mart)
mart &lt;- useMart(&quot;ensembl&quot;)
listDatasets(mart)
ensembl &lt;- useMart(&quot;ensembl&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;, host=&quot;grch37.ensembl.org&quot;, path=&quot;/biomart/martservice&quot;)
attr &lt;- listAttributes(ensembl)
attr_select &lt;- c('ensembl_gene_id', 'chromosome_name', 'start_position', 'end_position', 'description', 'hgnc_symbol', 'transcription_start_site')
gene &lt;- getBM(attributes = attr_select, mart = ensembl)
filter &lt;- listFilters(ensembl)
searchFilters(mart = ensembl, pattern = &quot;gene&quot;)
</code></pre>
<p>See also <a href="https://sites.google.com/site/jpopgen/wgsa">https://sites.google.com/site/jpopgen/wgsa</a> for precompiled annotation. Alternatively, </p>
<pre><code class="language-r"># GENCODE v19
url &lt;- &quot;ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_19/gencode.v19.chr_patch_hapl_scaff.annotation.gtf.gz&quot;
gtf &lt;- rtracklayer::import(url)
gencode &lt;- as.data.frame(gtf)
</code></pre>
<h2 id="biobanks">Biobanks</h2>
<ul>
<li>China Kadoorie Biobank<ul>
<li><a href="https://www.ckbiobank.org/">https://www.ckbiobank.org/</a></li>
<li><a href="https://pheweb.ckbiobank.org/">https://pheweb.ckbiobank.org/</a></li>
</ul>
</li>
<li>Estonian Biobank<ul>
<li><a href="https://genomics.ut.ee/en/">https://genomics.ut.ee/en/</a></li>
</ul>
</li>
<li>FinGenn<ul>
<li><a href="https://www.finngen.fi/en">https://www.finngen.fi/en</a></li>
</ul>
</li>
<li>Japan Biobank<ul>
<li><a href="https://biobankjp.org/">https://biobankjp.org/</a></li>
<li><a href="https://humandbs.biosciencedbc.jp/en/hum0014-v21">https://humandbs.biosciencedbc.jp/en/hum0014-v21</a></li>
</ul>
</li>
<li>UK Biobank<ul>
<li><a href="https://ams.ukbiobank.ac.uk/ams/">AMS</a> (<a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#97;&#99;&#99;&#101;&#115;&#115;&#64;&#117;&#107;&#98;&#105;&#111;&#98;&#97;&#110;&#107;&#46;&#97;&#99;&#46;&#117;&#107;">&#97;&#99;&#99;&#101;&#115;&#115;&#64;&#117;&#107;&#98;&#105;&#111;&#98;&#97;&#110;&#107;&#46;&#97;&#99;&#46;&#117;&#107;</a>), <a href="https://www.ukbiobank.ac.uk/wp-content/uploads/2019/09/Access_019-Access-Management-System-User-Guide-V4.0.pdf">Access_019-Access-Management-System-User-Guide-V4.0.pdf</a>, <a href="https://bbams.ndph.ox.ac.uk/ams/resMessages">messages</a>.</li>
<li>Accessing data guide, <a href="http://biobank.ctsu.ox.ac.uk/crystal/exinfo.cgi?src=AccessingData">http://biobank.ctsu.ox.ac.uk/crystal/exinfo.cgi?src=AccessingData</a>.</li>
<li>Allele frequency browser, <a href="https://afb.ukbiobank.ac.uk/">https://afb.ukbiobank.ac.uk/</a>.</li>
<li>AstraZeneca PheWAS Portal, <a href="https://azphewas.com/">https://azphewas.com/</a> (<a href="https://astrazeneca-cgr-publications.github.io/pqtl-browser/">CGR Proteogenomics Portal</a>).</li>
<li>Data access guide 3.2, <a href="https://biobank.ndph.ox.ac.uk/~bbdatan/Data_Access_Guide_v3.2.pdf">https://biobank.ndph.ox.ac.uk/~bbdatan/Data_Access_Guide_v3.2.pdf</a>.</li>
<li>DNAnexus, <a href="https://github.com/dnanexus">GitHub</a>, <a href="https://ukbiobank.dnanexus.com/landing">landing</a>, <a href="https://www.dnanexus.com/partnerships/ukbiobank">partnerships</a></li>
<li>Evoker, <a href="https://www.sanger.ac.uk/tool/evoker/">https://www.sanger.ac.uk/tool/evoker/</a>.</li>
<li>Genetic correlation between traits and disorders, <a href="https://ukbb-rg.hail.is/">https://ukbb-rg.hail.is/</a>.1;5C</li>
<li>Imputation, <a href="http://biobank.ctsu.ox.ac.uk/crystal/crystal/docs/impute_ukb_v1.pdf">http://biobank.ctsu.ox.ac.uk/crystal/crystal/docs/impute_ukb_v1.pdf</a>.</li>
<li>Gene ATLAS, <a href="http://geneatlas.roslin.ed.ac.uk/">http://geneatlas.roslin.ed.ac.uk/</a>.</li>
<li><a href="https://genebass.org/">genebass</a>.</li>
<li>UKB-Biobank, <a href="https://github.com/UK-Biobank">https://github.com/UK-Biobank</a> (<a href="https://github.com/lcpilling/ukbrapR">ukbrapR</a>, <a href="https://cran.r-project.org/package=ukbnmr">ukbnmr</a>)</li>
<li><a href="https://github.com/MRCIEU/PHESANT">PHESANT</a>.</li>
<li>Online showcase, <a href="https://biobank.ndph.ox.ac.uk/ukb/">https://biobank.ndph.ox.ac.uk/ukb/</a> (<a href="https://biobank.ctsu.ox.ac.uk/crystal/crystal/exinfo/ShowcaseUserGuide.pdf">Showcase User Guide</a>).</li>
<li><a href="https://pan.ukbb.broadinstitute.org/">Pan-UK Biobank</a>, <a href="http://www.nealelab.is/blog/2019/9/16/biomarkers-gwas-results">GWAS sumstats</a> and <a href="https://github.com/Nealelab/UK_Biobank_GWAS">GitHub</a>.</li>
<li>COVID-19 <a href="http://biobank.ndph.ox.ac.uk/ukb/exinfo.cgi?src=COVID19">data</a>, <a href="http://biobank.ndph.ox.ac.uk/ukb/exinfo.cgi?src=COVID19_tests">format</a> and <a href="http://biobank.ctsu.ox.ac.uk/ukb/field.cgi?id=40100">field</a>.</li>
</ul>
</li>
</ul>
<h2 id="catalog">Catalog</h2>
<ul>
<li>eQTL Catalog, <a href="https://www.ebi.ac.uk/eqtl/">https://www.ebi.ac.uk/eqtl/</a></li>
<li>GWAS Catalog, <a href="https://www.ebi.ac.uk/gwas/">https://www.ebi.ac.uk/gwas/</a></li>
<li>PheWAS Catalog, <a href="https://phewascatalog.org/">https://phewascatalog.org/</a></li>
</ul>
<h2 id="efo">EFO</h2>
<p><a href="https://www.ebi.ac.uk/efo/">https://www.ebi.ac.uk/efo/</a></p>
<p>Example code,</p>
<pre><code class="language-r">library(ontologyIndex)

id &lt;- function(ontology)
{
  inflammatory &lt;- grep(ontology$name,pattern=&quot;inflammatory&quot;)
  immune &lt;- grep(ontology$name,pattern=&quot;immune&quot;)
  inf &lt;- union(inflammatory,immune)
  list(id=ontology$id[inf],name=ontology$name[inf])
}
# GO
data(go)
goidname &lt;- id(go)
# EFO
file &lt;- &quot;efo.obo&quot;
get_relation_names(file)
efo &lt;- get_ontology(file, extract_tags=&quot;everything&quot;)
length(efo) # 89
length(efo$id) # 27962
efoidname &lt;- id(efo)
diseases &lt;- get_descendants(efo,&quot;EFO:0000408&quot;)
efo_0000540 &lt;- get_descendants(efo,&quot;EFO:0000540&quot;)
efo_0000540name &lt;- efo$name[efo_0000540]
isd &lt;- data.frame(efo_0000540,efo_0000540name)
save(efo,diseases,isd,efoidname,goidname, file=&quot;work/efo.rda&quot;)
write.table(isd,file=&quot;efo_0000540.csv&quot;,col.names=FALSE,row.names=FALSE,sep=&quot;,&quot;)
pdf(&quot;efo_0000540.pdf&quot;,height=15,width=15)
library(ontologyPlot)
onto_plot(efo,efo_0000540)
dev.off()
</code></pre>
<h2 id="eqtlgen">eQTLGen</h2>
<p><a href="http://www.eqtlgen.org/">http://www.eqtlgen.org/</a></p>
<h2 id="metamaplite">MetaMapLite</h2>
<p><a href="https://metamap.nlm.nih.gov/MetaMapLite.shtml">https://metamap.nlm.nih.gov/MetaMapLite.shtml</a></p>
<h2 id="mr-baseopengwas">MR-Base/OpenGWAS</h2>
<ul>
<li><a href="http://www.mrbase.org">http://www.mrbase.org</a> (<a href="https://github.com/MRCIEU/ieu-gwas-db-demo/">MRCIEU demo</a>)</li>
<li><a href="https://gwas.mrcieu.ac.uk">https://gwas.mrcieu.ac.uk</a></li>
</ul>
<h2 id="opentargets">OpenTargets</h2>
<p>These reflects v4 using GraphQL, <a href="https://platform-docs.opentargets.org/data-access/graphql-api">https://platform-docs.opentargets.org/data-access/graphql-api</a>.</p>
<p>Our first example is from the document (except ENSG00000164308) whose output is as <code>ERAP2.json</code>.</p>
<pre><code class="language-bash">#!/usr/bin/bash

# https://platform.opentargets.org/api

module load ceuadmin/R
Rscript -e '
  ERAP2 &lt;- subset(pQTLdata::caprion,Gene==&quot;ERAP2&quot;)
  ERAP2$ensGenes
  data &lt;- jsonlite::fromJSON(&quot;ERAP2.json&quot;)
  diseases_data &lt;- data$data$target$associatedDiseases$rows
  diseases_data &lt;- tidyr::unnest(diseases_data, cols = everything(), names_sep = &quot;_&quot;)
  write.table(diseases_data, file = &quot;ERAP2.tsv&quot;, sep = &quot;\t&quot;, row.names = FALSE, quote = FALSE)
'
</code></pre>
<p>In fact it is effectively done as follows,</p>
<pre><code class="language-r">library(httr)
library(jsonlite)
gene_id &lt;- &quot;ENSG00000164308&quot;
query_string &lt;- &quot;
  query target($ensemblId: String!){
    target(ensemblId: $ensemblId){
      id
      approvedSymbol
      associatedDiseases {
        count
        rows {
          disease {
            id
            name
          }
          datasourceScores {
            id
            score
          }
        }
      }
    }
  }
&quot;
base_url &lt;- &quot;https://api.platform.opentargets.org/api/v4/graphql&quot;
variables &lt;- list(&quot;ensemblId&quot; = gene_id)
post_body &lt;- list(query = query_string, variables = variables)
r &lt;- httr::POST(url = base_url, body = post_body, encode = 'json')
if (status_code(r) == 200) {
  data &lt;- iconv(r, &quot;&quot;, &quot;ASCII&quot;)
  content &lt;- jsonlite::fromJSON(data)
} else {
  print(paste(&quot;Request failed with status code&quot;, status_code(r)))
}
# Step 1: Access the nested data
target &lt;- content$data$target

# Step 2: Extract scalar fields
scalar_fields &lt;- data.frame(
  Field = c(&quot;ID&quot;, &quot;Approved Symbol&quot;, &quot;Biotype&quot;),
  Value = c(target$id, target$approvedSymbol, target$biotype)
)

# Step 3: Generate a table for scalar fields
cat(&quot;### Scalar Fields\n&quot;)
knitr::kable(scalar_fields, caption = &quot;Basic Information&quot;)

# Step 4: Generate a table for geneticConstraint
cat(&quot;\n### Genetic Constraint\n&quot;)
knitr::kable(target$geneticConstraint, caption = &quot;Genetic Constraint Metrics&quot;)

# Step 5: Generate a table for tractability
cat(&quot;\n### Tractability\n&quot;)
knitr::kable(target$tractability, caption = &quot;Tractability Information&quot;)
</code></pre>
<p>where <code>jsonlite::fromJSON(content(r,"text"))</code> is also possible when R is nicely compiled with libiconv.</p>
<p>A Bash implementation is copied here</p>
<pre><code class="language-bash">curl 'https://api.platform.opentargets.org/api/v4/graphql' \
     -H 'Accept-Encoding: gzip, deflate, br' \
     -H 'Content-Type: application/json' \
     -H 'Accept: application/json' \
     -H 'Connection: keep-alive' \
     -H 'DNT: 1' \
     -H 'Origin: https://api.platform.opentargets.org' \
     --data-binary '{&quot;query&quot;:&quot;query targetInfo {\n  target(ensemblId: \&quot;ENSG00000164308\&quot;) {\n    id\n    approvedSymbol\n    biotype\n    geneticConstraint {\n      constraintType\n      exp\n      obs\n      score\n      oe\n      oeLower\n      oeUpper\n    }\n    tractability {\n      label\n      modality\n      value\n    }\n  }\n}\n&quot;}' \
     --compressed
</code></pre>
<p>The Python script can be used directly without change</p>
<pre><code class="language-python3">import requests
import json

gene_id = &quot;ENSG00000164308&quot;
query_string = &quot;&quot;&quot;
  query target($ensemblId: String!){
    target(ensemblId: $ensemblId){
      id
      approvedSymbol
      biotype
      geneticConstraint {
        constraintType
        exp
        obs
        score
        oe
        oeLower
        oeUpper
      }
      tractability {
        label
        modality
        value
      }
    }
  }
&quot;&quot;&quot;
variables = {&quot;ensemblId&quot;: gene_id}
base_url = &quot;https://api.platform.opentargets.org/api/v4/graphql&quot;
r = requests.post(base_url, json={&quot;query&quot;: query_string, &quot;variables&quot;: variables})
print(r.status_code)
api_response = json.loads(r.text)
print(api_response)
</code></pre>
<p>Lastly we turn to R, which again gets around <code>httr::content(r)</code> for <code>iconvlist()</code> with <code>iconv()</code>.</p>
<pre><code class="language-r">library(httr)
library(jsonlite)

gene_id &lt;- &quot;ENSG00000164308&quot;
query_string = &quot;
  query target($ensemblId: String!){
    target(ensemblId: $ensemblId){
      id
      approvedSymbol
      biotype
      geneticConstraint {
        constraintType
        exp
        obs
        score
        oe
        oeLower
        oeUpper
      }
      tractability {
        label
        modality
        value
      }
    }
  }
&quot;
base_url &lt;- &quot;https://api.platform.opentargets.org/api/v4/graphql&quot;
variables &lt;- list(&quot;ensemblId&quot; = gene_id)
post_body &lt;- list(query = query_string, variables = variables)
r &lt;- httr::POST(url=base_url, body=post_body, encode='json')
data &lt;- iconv(r, &quot;&quot;, &quot;ASCII&quot;)
content &lt;- jsonlite::fromJSON(data)
</code></pre>
<h2 id="rentrez">rentrez</h2>
<p>The relevant URLs are as follows, </p>
<ul>
<li><a href="https://cran.r-project.org/web/packages/rentrez/vignettes/rentrez_tutorial.html">https://cran.r-project.org/web/packages/rentrez/vignettes/rentrez_tutorial.html</a></li>
<li><a href="https://pubmed.ncbi.nlm.nih.gov/">https://pubmed.ncbi.nlm.nih.gov/</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/pmc/pmctopmid/">https://www.ncbi.nlm.nih.gov/pmc/pmctopmid/</a></li>
</ul>
<p>with example code,</p>
<pre><code class="language-r">library(rentrez)
entrez_dbs()
entrez_db_links(&quot;pubmed&quot;)
pubmed_fields &lt;- entrez_db_searchable(&quot;pubmed&quot;)
# set_entrez_key(&quot;&quot;)
Sys.getenv(&quot;ENTREZ_KEY&quot;)
term &lt;- &quot;pQTLs OR (protein AND quantitative AND trait AND loci) AND human [MH] AND (plasma OR Serum)&quot;
r &lt;- entrez_search(db=&quot;pubmed&quot;,term=term,use_history=TRUE)
class(r)
names(r)
with(r,web_history)
unlink(paste(&quot;pubmed&quot;,c(&quot;fetch&quot;,&quot;summary&quot;),sep=&quot;.&quot;))
fields &lt;- c(&quot;uid&quot;, &quot;pubdate&quot;, &quot;sortfirstauthor&quot;, &quot;title&quot;, &quot;source&quot;, &quot;volume&quot;, &quot;pages&quot;)
for(i in seq(1,with(r,count),50))
{
  cat(i+49, &quot;records downloaded\r&quot;)
  f &lt;- entrez_fetch(db=&quot;pubmed&quot;, web_history=with(r,web_history), rettype=&quot;text&quot;, retmax=50, retstart=i)
  write.table(f, col.names=FALSE, row.names=FALSE, file=&quot;pubmed.fetch&quot;, append=TRUE)
  s &lt;- entrez_summary(db=&quot;pubmed&quot;, web_history=with(r,web_history), rettype=&quot;text&quot;, retmax=50, retstart=i)
  e &lt;- extract_from_esummary(s, fields)
  write.table(t(e), col.names=FALSE, row.names=FALSE, file=&quot;pubmed.summary&quot;, append=TRUE, sep=&quot;\t&quot;)
}
id &lt;- 600807
upload &lt;- entrez_post(db=&quot;omim&quot;, id=id)
asthma_variants &lt;- entrez_link(dbfrom=&quot;omim&quot;, db=&quot;clinvar&quot;, cmd=&quot;neighbor_history&quot;, web_history=upload)
asthma_variants
snp_links &lt;- entrez_link(dbfrom=&quot;clinvar&quot;, db=&quot;snp&quot;, web_history=asthma_variants$web_histories$omim_clinvar, cmd=&quot;neighbor_history&quot;)
all_links &lt;- entrez_link(dbfrom='pubmed', id=id, db='all')
</code></pre>
<h2 id="roadmap">Roadmap</h2>
<p><a href="http://www.roadmapepigenomics.org/">http://www.roadmapepigenomics.org/</a></p>
<h2 id="snakemake-workflow-catalogue">snakemake workflow catalogue</h2>
<p><a href="https://snakemake.github.io/snakemake-workflow-catalog/">https://snakemake.github.io/snakemake-workflow-catalog/</a></p>
<h2 id="twas">TWAS</h2>
<ul>
<li>MetaXcan, <a href="https://github.com/hakyimlab/MetaXcan">https://github.com/hakyimlab/MetaXcan</a></li>
<li>FUSION, <a href="http://gusevlab.org/projects/fusion/">http://gusevlab.org/projects/fusion/</a></li>
<li>OmicsPred, <a href="https://www.omicspred.org/">https://www.omicspred.org/</a></li>
<li>PredictDB data repository, <a href="http://predictdb.org/">http://predictdb.org/</a></li>
<li>TWAS-hub, <a href="http://twas-hub.org/">http://twas-hub.org/</a></li>
</ul>
<h2 id="other-links">Other links</h2>
<ul>
<li><a href="https://www.ucl.ac.uk/health-informatics/caliber">CALIBER</a>.</li>
<li>deCode <a href="https://www.decode.com/summarydata/">summary statistics</a></li>
<li><a href="https://usegalaxy.eu/">Galaxy Europe</a></li>
<li><a href="https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/giab/data/">Genome in a Bottle WGS samples</a></li>
<li><a href="https://portal.genego.com/">genego</a>.</li>
<li>MVP GWAS summary statistics, <a href="https://ftp.ncbi.nlm.nih.gov/dbgap/studies/phs002453/analyses/">https://ftp.ncbi.nlm.nih.gov/dbgap/studies/phs002453/analyses/</a>.</li>
<li><a href="https://github.com/aehrc">The Australian e-Health Research Centre</a>.</li>
<li><a href="https://github.com/hmgu-itg">Institute of Translational Genomics</a> and <a href="https://omicscience.org/">omicscience</a>.</li>
<li><a href="http://bioinformatics.sdstate.edu/idep/">idep</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/myncbi/">NCBI account</a> (<a href="https://www.ncbi.nlm.nih.gov/account/settings/">settings</a>).</li>
</ul></div>
            </div>
        </div>

        <footer class="col-md-12">
            <hr>
            <p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>
        <script src="../js/jquery-3.6.0.min.js"></script>
        <script src="../js/bootstrap.min.js"></script>
        <script>
            var base_url = "..",
                shortcuts = {"help": 191, "next": 78, "previous": 80, "search": 83};
        </script>
        <script src="../js/base.js"></script>
        <script src="../javascripts/config.js"></script>
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
        <script src="../search/main.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="searchModalLabel" aria-hidden="true">
    <div class="modal-dialog modal-lg">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="searchModalLabel">Search</h4>
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            </div>
            <div class="modal-body">
                <p>From here you can search these documents. Enter your search terms below.</p>
                <form>
                    <div class="form-group">
                        <input type="search" class="form-control" placeholder="Search..." id="mkdocs-search-query" title="Type search term here">
                    </div>
                </form>
                <div id="mkdocs-search-results" data-no-results-text="No results found"></div>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div><div class="modal" id="mkdocs_keyboard_modal" tabindex="-1" role="dialog" aria-labelledby="keyboardModalLabel" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="keyboardModalLabel">Keyboard Shortcuts</h4>
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            </div>
            <div class="modal-body">
              <table class="table">
                <thead>
                  <tr>
                    <th style="width: 20%;">Keys</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td class="help shortcut"><kbd>?</kbd></td>
                    <td>Open this help</td>
                  </tr>
                  <tr>
                    <td class="next shortcut"><kbd>n</kbd></td>
                    <td>Next page</td>
                  </tr>
                  <tr>
                    <td class="prev shortcut"><kbd>p</kbd></td>
                    <td>Previous page</td>
                  </tr>
                  <tr>
                    <td class="search shortcut"><kbd>s</kbd></td>
                    <td>Search</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div>

    </body>
</html>
