<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="../img/favicon.ico">
        <title>Resources - Omics-analysis</title>
        <link href="../css/bootstrap.min.css" rel="stylesheet">
        <link href="../css/font-awesome.min.css" rel="stylesheet">
        <link href="../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css">
        <link href="../stylesheets/extra.css" rel="stylesheet">
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/Bash.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/HTML.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/Java.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/JavaScript.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/Perl.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/Python.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/R.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/rust.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/yaml.min.js"></script>
        <script>hljs.highlightAll();</script> 
    </head>

    <body>
        <div class="navbar fixed-top navbar-expand-lg navbar-dark bg-dark">
            <div class="container">
                <a class="navbar-brand" href="..">Omics-analysis</a>
                <!-- Expander button -->
                <button type="button" class="navbar-toggler" data-toggle="collapse" data-target="#navbar-collapse">
                    <span class="navbar-toggler-icon"></span>
                </button>

                <!-- Expanded navigation -->
                <div id="navbar-collapse" class="navbar-collapse collapse">
                        <!-- Main navigation -->
                        <ul class="nav navbar-nav">
                            <li class="navitem">
                                <a href=".." class="nav-link"></a>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">Cases <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li>
    <a href="../AF/" class="dropdown-item">Atrial Fibrillation</a>
</li>
                                    
<li>
    <a href="../BMI/" class="dropdown-item">BMI</a>
</li>
                                    
<li>
    <a href="../CAD/" class="dropdown-item">CAD/MI</a>
</li>
                                    
<li>
    <a href="../T2D/" class="dropdown-item">T2D</a>
</li>
                                </ul>
                            </li>
                            <li class="navitem">
                                <a href="../genomics/" class="nav-link">Genomics</a>
                            </li>
                            <li class="navitem">
                                <a href="../proteomics/" class="nav-link">Proteomics</a>
                            </li>
                            <li class="navitem active">
                                <a href="./" class="nav-link">Resources</a>
                            </li>
                        </ul>

                    <ul class="nav navbar-nav ml-auto">
                        <li class="nav-item">
                            <a href="#" class="nav-link" data-toggle="modal" data-target="#mkdocs_search_modal">
                                <i class="fa fa-search"></i> Search
                            </a>
                        </li>
                            <li class="nav-item">
                                <a rel="prev" href="../proteomics/" class="nav-link">
                                    <i class="fa fa-arrow-left"></i> Previous
                                </a>
                            </li>
                            <li class="nav-item">
                                <a rel="next" class="nav-link disabled">
                                    Next <i class="fa fa-arrow-right"></i>
                                </a>
                            </li>
                            <li class="nav-item">
                                <a href="https://github.com/jinghuazhao/Omics-analysis" class="nav-link"><i class="fa fa-github"></i> GitHub</a>
                            </li>
                    </ul>
                </div>
            </div>
        </div>

        <div class="container">
            <div class="row">
                    <div class="col-md-3"><div class="navbar-light navbar-expand-md bs-sidebar hidden-print affix" role="complementary">
    <div class="navbar-header">
        <button type="button" class="navbar-toggler collapsed" data-toggle="collapse" data-target="#toc-collapse" title="Table of Contents">
            <span class="fa fa-angle-down"></span>
        </button>
    </div>

    
    <div id="toc-collapse" class="navbar-collapse collapse card bg-secondary">
        <ul class="nav flex-column">
            
            <li class="nav-item" data-level="1"><a href="#resources" class="nav-link">Resources</a>
              <ul class="nav flex-column">
            <li class="nav-item" data-level="2"><a href="#annotation" class="nav-link">Annotation</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#biobanks" class="nav-link">Biobanks</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#catalog" class="nav-link">Catalog</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#efo" class="nav-link">EFO</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#eqtlgen" class="nav-link">eQTLGen</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#metamaplite" class="nav-link">MetaMapLite</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#mr-baseopengwas" class="nav-link">MR-Base/OpenGWAS</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#opentargets" class="nav-link">OpenTargets</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#pgs-searchgpt" class="nav-link">PGS (searchGPT)</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#rentrez" class="nav-link">rentrez</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#roadmap" class="nav-link">Roadmap</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#snakemake-workflow-catalogue" class="nav-link">snakemake workflow catalogue</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#twas" class="nav-link">TWAS</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#uk10k-imputation" class="nav-link">UK10K imputation</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#other-links" class="nav-link">Other links</a>
              <ul class="nav flex-column">
              </ul>
            </li>
              </ul>
            </li>
        </ul>
    </div>
</div></div>
                    <div class="col-md-9" role="main">

<h1 id="resources">Resources</h1>
<h2 id="annotation">Annotation</h2>
<p><a href="https://www.ensembl.org/info/data/mysql.html">The Ensembl public MySQL Servers</a></p>
<p>The following script gives information on genes from ENSEMBL as well as attributes (columns) that contains <code>gene</code>.
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">biomaRt</span><span class="p">)</span>
<span class="nf">listMarts</span><span class="p">()</span>
<span class="n">mart</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">useMart</span><span class="p">(</span><span class="s">&quot;ENSEMBL_MART_FUNCGEN&quot;</span><span class="p">)</span>
<span class="nf">listDatasets</span><span class="p">(</span><span class="n">mart</span><span class="p">)</span>
<span class="n">mart</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">useMart</span><span class="p">(</span><span class="s">&quot;ensembl&quot;</span><span class="p">)</span>
<span class="nf">listDatasets</span><span class="p">(</span><span class="n">mart</span><span class="p">)</span>
<span class="n">ensembl</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">useMart</span><span class="p">(</span><span class="s">&quot;ensembl&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">dataset</span><span class="o">=</span><span class="s">&quot;hsapiens_gene_ensembl&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">host</span><span class="o">=</span><span class="s">&quot;grch37.ensembl.org&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">path</span><span class="o">=</span><span class="s">&quot;/biomart/martservice&quot;</span><span class="p">)</span>
<span class="n">attr</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">listAttributes</span><span class="p">(</span><span class="n">ensembl</span><span class="p">)</span>
<span class="n">attr_select</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;ensembl_gene_id&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;chromosome_name&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;start_position&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;end_position&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;description&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;hgnc_symbol&#39;</span><span class="p">,</span><span class="w"> </span><span class="s">&#39;transcription_start_site&#39;</span><span class="p">)</span>
<span class="n">gene</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">getBM</span><span class="p">(</span><span class="n">attributes</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">attr_select</span><span class="p">,</span><span class="w"> </span><span class="n">mart</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ensembl</span><span class="p">)</span>
<span class="n">filter</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">listFilters</span><span class="p">(</span><span class="n">ensembl</span><span class="p">)</span>
<span class="nf">searchFilters</span><span class="p">(</span><span class="n">mart</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">ensembl</span><span class="p">,</span><span class="w"> </span><span class="n">pattern</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;gene&quot;</span><span class="p">)</span>
</code></pre></div>
See also <a href="https://sites.google.com/site/jpopgen/wgsa">https://sites.google.com/site/jpopgen/wgsa</a> for precompiled annotation. Alternatively, 
<div class="highlight"><pre><span></span><code><span class="c1"># GENCODE v19</span>
<span class="n">url</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_19/gencode.v19.chr_patch_hapl_scaff.annotation.gtf.gz&quot;</span>
<span class="n">gtf</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rtracklayer</span><span class="o">::</span><span class="nf">import</span><span class="p">(</span><span class="n">url</span><span class="p">)</span>
<span class="n">gencode</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">as.data.frame</span><span class="p">(</span><span class="n">gtf</span><span class="p">)</span>
</code></pre></div></p>
<h2 id="biobanks">Biobanks</h2>
<ul>
<li>China Kadoorie Biobank<ul>
<li><a href="https://www.ckbiobank.org/">https://www.ckbiobank.org/</a></li>
<li><a href="https://pheweb.ckbiobank.org/">https://pheweb.ckbiobank.org/</a></li>
</ul>
</li>
<li>Estonian Biobank<ul>
<li><a href="https://genomics.ut.ee/en/">https://genomics.ut.ee/en/</a></li>
</ul>
</li>
<li>FinGenn<ul>
<li><a href="https://www.finngen.fi/en">https://www.finngen.fi/en</a></li>
</ul>
</li>
<li>Japan Biobank<ul>
<li><a href="https://biobankjp.org/">https://biobankjp.org/</a></li>
<li><a href="https://humandbs.biosciencedbc.jp/en/hum0014-v21">https://humandbs.biosciencedbc.jp/en/hum0014-v21</a></li>
</ul>
</li>
<li>Our Future Health, <a href="https://research.ourfuturehealth.org.uk/">https://research.ourfuturehealth.org.uk/</a></li>
<li>UK Biobank<ul>
<li><a href="https://ams.ukbiobank.ac.uk/ams/">AMS</a> (<a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#97;&#99;&#99;&#101;&#115;&#115;&#64;&#117;&#107;&#98;&#105;&#111;&#98;&#97;&#110;&#107;&#46;&#97;&#99;&#46;&#117;&#107;">&#97;&#99;&#99;&#101;&#115;&#115;&#64;&#117;&#107;&#98;&#105;&#111;&#98;&#97;&#110;&#107;&#46;&#97;&#99;&#46;&#117;&#107;</a>), <a href="https://www.ukbiobank.ac.uk/wp-content/uploads/2019/09/Access_019-Access-Management-System-User-Guide-V4.0.pdf">Access_019-Access-Management-System-User-Guide-V4.0.pdf</a>, <a href="https://bbams.ndph.ox.ac.uk/ams/resMessages">messages</a>.</li>
<li>Accessing data guide, <a href="http://biobank.ctsu.ox.ac.uk/crystal/exinfo.cgi?src=AccessingData">http://biobank.ctsu.ox.ac.uk/crystal/exinfo.cgi?src=AccessingData</a>.</li>
<li>Allele frequency browser, <a href="https://afb.ukbiobank.ac.uk/">https://afb.ukbiobank.ac.uk/</a>.</li>
<li>AstraZeneca PheWAS Portal, <a href="https://azphewas.com/">https://azphewas.com/</a> (<a href="https://astrazeneca-cgr-publications.github.io/pqtl-browser/">CGR Proteogenomics Portal</a>).</li>
<li>Data access guide 3.2, <a href="https://biobank.ndph.ox.ac.uk/~bbdatan/Data_Access_Guide_v3.2.pdf">https://biobank.ndph.ox.ac.uk/~bbdatan/Data_Access_Guide_v3.2.pdf</a>.</li>
<li>DNAnexus, <a href="https://github.com/dnanexus">GitHub</a>, <a href="https://ukbiobank.dnanexus.com/landing">landing</a>, <a href="https://www.dnanexus.com/partnerships/ukbiobank">partnerships</a></li>
<li>Evoker, <a href="https://www.sanger.ac.uk/tool/evoker/">https://www.sanger.ac.uk/tool/evoker/</a>.</li>
<li>Genetic correlation between traits and disorders, <a href="https://ukbb-rg.hail.is/">https://ukbb-rg.hail.is/</a>.</li>
<li>Imputation, <a href="http://biobank.ctsu.ox.ac.uk/crystal/crystal/docs/impute_ukb_v1.pdf">http://biobank.ctsu.ox.ac.uk/crystal/crystal/docs/impute_ukb_v1.pdf</a>.</li>
<li>Gene ATLAS, <a href="http://geneatlas.roslin.ed.ac.uk/">http://geneatlas.roslin.ed.ac.uk/</a>.</li>
<li><a href="https://genebass.org/">genebass</a>.</li>
<li>UKB-Biobank, <a href="https://github.com/UK-Biobank">https://github.com/UK-Biobank</a> (<a href="https://github.com/lcpilling/ukbrapR">ukbrapR</a>, <a href="https://cran.r-project.org/package=ukbnmr">ukbnmr</a>)</li>
<li><a href="https://github.com/MRCIEU/PHESANT">PHESANT</a>.</li>
<li>Online showcase, <a href="https://biobank.ndph.ox.ac.uk/ukb/">https://biobank.ndph.ox.ac.uk/ukb/</a> (<a href="https://biobank.ctsu.ox.ac.uk/crystal/crystal/exinfo/ShowcaseUserGuide.pdf">Showcase User Guide</a>).</li>
<li><a href="https://pan.ukbb.broadinstitute.org/">Pan-UK Biobank</a>, <a href="http://www.nealelab.is/blog/2019/9/16/biomarkers-gwas-results">GWAS sumstats</a> and <a href="https://github.com/Nealelab/UK_Biobank_GWAS">GitHub</a>.</li>
<li>COVID-19 <a href="http://biobank.ndph.ox.ac.uk/ukb/exinfo.cgi?src=COVID19">data</a>, <a href="http://biobank.ndph.ox.ac.uk/ukb/exinfo.cgi?src=COVID19_tests">format</a> and <a href="http://biobank.ctsu.ox.ac.uk/ukb/field.cgi?id=40100">field</a>.</li>
</ul>
</li>
</ul>
<h2 id="catalog">Catalog</h2>
<ul>
<li>eQTL Catalog, <a href="https://www.ebi.ac.uk/eqtl/">https://www.ebi.ac.uk/eqtl/</a></li>
<li>GWAS Catalog, <a href="https://www.ebi.ac.uk/gwas/">https://www.ebi.ac.uk/gwas/</a></li>
<li>PheWAS Catalog, <a href="https://phewascatalog.org/">https://phewascatalog.org/</a></li>
</ul>
<h2 id="efo">EFO</h2>
<p><a href="https://www.ebi.ac.uk/efo/">https://www.ebi.ac.uk/efo/</a></p>
<p>Example code,
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">ontologyIndex</span><span class="p">)</span>

<span class="n">id</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">function</span><span class="p">(</span><span class="n">ontology</span><span class="p">)</span>
<span class="p">{</span>
<span class="w">  </span><span class="n">inflammatory</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">grep</span><span class="p">(</span><span class="n">ontology</span><span class="o">$</span><span class="n">name</span><span class="p">,</span><span class="n">pattern</span><span class="o">=</span><span class="s">&quot;inflammatory&quot;</span><span class="p">)</span>
<span class="w">  </span><span class="n">immune</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">grep</span><span class="p">(</span><span class="n">ontology</span><span class="o">$</span><span class="n">name</span><span class="p">,</span><span class="n">pattern</span><span class="o">=</span><span class="s">&quot;immune&quot;</span><span class="p">)</span>
<span class="w">  </span><span class="n">inf</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">union</span><span class="p">(</span><span class="n">inflammatory</span><span class="p">,</span><span class="n">immune</span><span class="p">)</span>
<span class="w">  </span><span class="nf">list</span><span class="p">(</span><span class="n">id</span><span class="o">=</span><span class="n">ontology</span><span class="o">$</span><span class="n">id</span><span class="p">[</span><span class="n">inf</span><span class="p">],</span><span class="n">name</span><span class="o">=</span><span class="n">ontology</span><span class="o">$</span><span class="n">name</span><span class="p">[</span><span class="n">inf</span><span class="p">])</span>
<span class="p">}</span>
<span class="c1"># GO</span>
<span class="nf">data</span><span class="p">(</span><span class="n">go</span><span class="p">)</span>
<span class="n">goidname</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">id</span><span class="p">(</span><span class="n">go</span><span class="p">)</span>
<span class="c1"># EFO</span>
<span class="n">file</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;efo.obo&quot;</span>
<span class="nf">get_relation_names</span><span class="p">(</span><span class="n">file</span><span class="p">)</span>
<span class="n">efo</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">get_ontology</span><span class="p">(</span><span class="n">file</span><span class="p">,</span><span class="w"> </span><span class="n">extract_tags</span><span class="o">=</span><span class="s">&quot;everything&quot;</span><span class="p">)</span>
<span class="nf">length</span><span class="p">(</span><span class="n">efo</span><span class="p">)</span><span class="w"> </span><span class="c1"># 89</span>
<span class="nf">length</span><span class="p">(</span><span class="n">efo</span><span class="o">$</span><span class="n">id</span><span class="p">)</span><span class="w"> </span><span class="c1"># 27962</span>
<span class="n">efoidname</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">id</span><span class="p">(</span><span class="n">efo</span><span class="p">)</span>
<span class="n">diseases</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">get_descendants</span><span class="p">(</span><span class="n">efo</span><span class="p">,</span><span class="s">&quot;EFO:0000408&quot;</span><span class="p">)</span>
<span class="n">efo_0000540</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">get_descendants</span><span class="p">(</span><span class="n">efo</span><span class="p">,</span><span class="s">&quot;EFO:0000540&quot;</span><span class="p">)</span>
<span class="n">efo_0000540name</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">efo</span><span class="o">$</span><span class="n">name</span><span class="p">[</span><span class="n">efo_0000540</span><span class="p">]</span>
<span class="n">isd</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span><span class="n">efo_0000540</span><span class="p">,</span><span class="n">efo_0000540name</span><span class="p">)</span>
<span class="nf">save</span><span class="p">(</span><span class="n">efo</span><span class="p">,</span><span class="n">diseases</span><span class="p">,</span><span class="n">isd</span><span class="p">,</span><span class="n">efoidname</span><span class="p">,</span><span class="n">goidname</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="o">=</span><span class="s">&quot;work/efo.rda&quot;</span><span class="p">)</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">isd</span><span class="p">,</span><span class="n">file</span><span class="o">=</span><span class="s">&quot;efo_0000540.csv&quot;</span><span class="p">,</span><span class="n">col.names</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="n">row.names</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;,&quot;</span><span class="p">)</span>
<span class="nf">pdf</span><span class="p">(</span><span class="s">&quot;efo_0000540.pdf&quot;</span><span class="p">,</span><span class="n">height</span><span class="o">=</span><span class="m">15</span><span class="p">,</span><span class="n">width</span><span class="o">=</span><span class="m">15</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">ontologyPlot</span><span class="p">)</span>
<span class="nf">onto_plot</span><span class="p">(</span><span class="n">efo</span><span class="p">,</span><span class="n">efo_0000540</span><span class="p">)</span>
<span class="nf">dev.off</span><span class="p">()</span>
</code></pre></div></p>
<h2 id="eqtlgen">eQTLGen</h2>
<p><a href="http://www.eqtlgen.org/">http://www.eqtlgen.org/</a></p>
<h2 id="metamaplite">MetaMapLite</h2>
<p><a href="https://metamap.nlm.nih.gov/MetaMapLite.shtml">https://metamap.nlm.nih.gov/MetaMapLite.shtml</a></p>
<h2 id="mr-baseopengwas">MR-Base/OpenGWAS</h2>
<ul>
<li><a href="http://www.mrbase.org">http://www.mrbase.org</a> (<a href="https://github.com/MRCIEU/ieu-gwas-db-demo/">MRCIEU demo</a>)</li>
<li><a href="https://gwas.mrcieu.ac.uk">https://gwas.mrcieu.ac.uk</a></li>
</ul>
<h2 id="opentargets">OpenTargets</h2>
<p>These reflects v4 using GraphQL, <a href="https://platform-docs.opentargets.org/data-access/graphql-api">https://platform-docs.opentargets.org/data-access/graphql-api</a>.</p>
<p>Our first example is from the document (except ENSG00000164308) whose output is as <code>ERAP2.json</code>.</p>
<div class="highlight"><pre><span></span><code><span class="ch">#!/usr/bin/bash</span>

<span class="c1"># https://platform.opentargets.org/api</span>

module<span class="w"> </span>load<span class="w"> </span>ceuadmin/R
Rscript<span class="w"> </span>-e<span class="w"> </span><span class="s1">&#39;</span>
<span class="s1">  ERAP2 &lt;- subset(pQTLdata::caprion,Gene==&quot;ERAP2&quot;)</span>
<span class="s1">  ERAP2$ensGenes</span>
<span class="s1">  data &lt;- jsonlite::fromJSON(&quot;ERAP2.json&quot;)</span>
<span class="s1">  diseases_data &lt;- data$data$target$associatedDiseases$rows</span>
<span class="s1">  diseases_data &lt;- tidyr::unnest(diseases_data, cols = everything(), names_sep = &quot;_&quot;)</span>
<span class="s1">  write.table(diseases_data, file = &quot;ERAP2.tsv&quot;, sep = &quot;\t&quot;, row.names = FALSE, quote = FALSE)</span>
<span class="s1">&#39;</span>
</code></pre></div>
<p>In fact it is effectively done as follows,</p>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">httr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">jsonlite</span><span class="p">)</span>
<span class="n">gene_id</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;ENSG00000164308&quot;</span>
<span class="n">query_string</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;</span>
<span class="s">  query target($ensemblId: String!){</span>
<span class="s">    target(ensemblId: $ensemblId){</span>
<span class="s">      id</span>
<span class="s">      approvedSymbol</span>
<span class="s">      associatedDiseases {</span>
<span class="s">        count</span>
<span class="s">        rows {</span>
<span class="s">          disease {</span>
<span class="s">            id</span>
<span class="s">            name</span>
<span class="s">          }</span>
<span class="s">          datasourceScores {</span>
<span class="s">            id</span>
<span class="s">            score</span>
<span class="s">          }</span>
<span class="s">        }</span>
<span class="s">      }</span>
<span class="s">    }</span>
<span class="s">  }</span>
<span class="s">&quot;</span>
<span class="n">base_url</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;https://api.platform.opentargets.org/api/v4/graphql&quot;</span>
<span class="n">variables</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="s">&quot;ensemblId&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gene_id</span><span class="p">)</span>
<span class="n">post_body</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">query</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">query_string</span><span class="p">,</span><span class="w"> </span><span class="n">variables</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">variables</span><span class="p">)</span>
<span class="n">r</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">httr</span><span class="o">::</span><span class="nf">POST</span><span class="p">(</span><span class="n">url</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">base_url</span><span class="p">,</span><span class="w"> </span><span class="n">body</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">post_body</span><span class="p">,</span><span class="w"> </span><span class="n">encode</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&#39;json&#39;</span><span class="p">)</span>
<span class="nf">if </span><span class="p">(</span><span class="nf">status_code</span><span class="p">(</span><span class="n">r</span><span class="p">)</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">200</span><span class="p">)</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">iconv</span><span class="p">(</span><span class="n">r</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;ASCII&quot;</span><span class="p">)</span>
<span class="w">  </span><span class="n">content</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">jsonlite</span><span class="o">::</span><span class="nf">fromJSON</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
<span class="p">}</span><span class="w"> </span><span class="n">else</span><span class="w"> </span><span class="p">{</span>
<span class="w">  </span><span class="nf">print</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;Request failed with status code&quot;</span><span class="p">,</span><span class="w"> </span><span class="nf">status_code</span><span class="p">(</span><span class="n">r</span><span class="p">)))</span>
<span class="p">}</span>
<span class="c1"># Step 1: Access the nested data</span>
<span class="n">target</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">content</span><span class="o">$</span><span class="n">data</span><span class="o">$</span><span class="n">target</span>

<span class="c1"># Step 2: Extract scalar fields</span>
<span class="n">scalar_fields</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">data.frame</span><span class="p">(</span>
<span class="w">  </span><span class="n">Field</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;ID&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Approved Symbol&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;Biotype&quot;</span><span class="p">),</span>
<span class="w">  </span><span class="n">Value</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="n">target</span><span class="o">$</span><span class="n">id</span><span class="p">,</span><span class="w"> </span><span class="n">target</span><span class="o">$</span><span class="n">approvedSymbol</span><span class="p">,</span><span class="w"> </span><span class="n">target</span><span class="o">$</span><span class="n">biotype</span><span class="p">)</span>
<span class="p">)</span>

<span class="c1"># Step 3: Generate a table for scalar fields</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;### Scalar Fields\n&quot;</span><span class="p">)</span>
<span class="n">knitr</span><span class="o">::</span><span class="nf">kable</span><span class="p">(</span><span class="n">scalar_fields</span><span class="p">,</span><span class="w"> </span><span class="n">caption</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Basic Information&quot;</span><span class="p">)</span>

<span class="c1"># Step 4: Generate a table for geneticConstraint</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;\n### Genetic Constraint\n&quot;</span><span class="p">)</span>
<span class="n">knitr</span><span class="o">::</span><span class="nf">kable</span><span class="p">(</span><span class="n">target</span><span class="o">$</span><span class="n">geneticConstraint</span><span class="p">,</span><span class="w"> </span><span class="n">caption</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Genetic Constraint Metrics&quot;</span><span class="p">)</span>

<span class="c1"># Step 5: Generate a table for tractability</span>
<span class="nf">cat</span><span class="p">(</span><span class="s">&quot;\n### Tractability\n&quot;</span><span class="p">)</span>
<span class="n">knitr</span><span class="o">::</span><span class="nf">kable</span><span class="p">(</span><span class="n">target</span><span class="o">$</span><span class="n">tractability</span><span class="p">,</span><span class="w"> </span><span class="n">caption</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;Tractability Information&quot;</span><span class="p">)</span>
</code></pre></div>
<p>where <code>jsonlite::fromJSON(content(r,"text"))</code> is also possible when R is nicely compiled with libiconv.</p>
<p>A Bash implementation is copied here</p>
<div class="highlight"><pre><span></span><code>curl<span class="w"> </span><span class="s1">&#39;https://api.platform.opentargets.org/api/v4/graphql&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">     </span>-H<span class="w"> </span><span class="s1">&#39;Accept-Encoding: gzip, deflate, br&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">     </span>-H<span class="w"> </span><span class="s1">&#39;Content-Type: application/json&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">     </span>-H<span class="w"> </span><span class="s1">&#39;Accept: application/json&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">     </span>-H<span class="w"> </span><span class="s1">&#39;Connection: keep-alive&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">     </span>-H<span class="w"> </span><span class="s1">&#39;DNT: 1&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">     </span>-H<span class="w"> </span><span class="s1">&#39;Origin: https://api.platform.opentargets.org&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">     </span>--data-binary<span class="w"> </span><span class="s1">&#39;{&quot;query&quot;:&quot;query targetInfo {\n  target(ensemblId: \&quot;ENSG00000164308\&quot;) {\n    id\n    approvedSymbol\n    biotype\n    geneticConstraint {\n      constraintType\n      exp\n      obs\n      score\n      oe\n      oeLower\n      oeUpper\n    }\n    tractability {\n      label\n      modality\n      value\n    }\n  }\n}\n&quot;}&#39;</span><span class="w"> </span><span class="se">\</span>
<span class="w">     </span>--compressed
</code></pre></div>
<p>The Python script can be used directly without change</p>
<div class="highlight"><pre><span></span><code><span class="kn">import</span> <span class="nn">requests</span>
<span class="kn">import</span> <span class="nn">json</span>

<span class="n">gene_id</span> <span class="o">=</span> <span class="s2">&quot;ENSG00000164308&quot;</span>
<span class="n">query_string</span> <span class="o">=</span> <span class="s2">&quot;&quot;&quot;</span>
<span class="s2">  query target($ensemblId: String!){</span>
<span class="s2">    target(ensemblId: $ensemblId){</span>
<span class="s2">      id</span>
<span class="s2">      approvedSymbol</span>
<span class="s2">      biotype</span>
<span class="s2">      geneticConstraint {</span>
<span class="s2">        constraintType</span>
<span class="s2">        exp</span>
<span class="s2">        obs</span>
<span class="s2">        score</span>
<span class="s2">        oe</span>
<span class="s2">        oeLower</span>
<span class="s2">        oeUpper</span>
<span class="s2">      }</span>
<span class="s2">      tractability {</span>
<span class="s2">        label</span>
<span class="s2">        modality</span>
<span class="s2">        value</span>
<span class="s2">      }</span>
<span class="s2">    }</span>
<span class="s2">  }</span>
<span class="s2">&quot;&quot;&quot;</span>
<span class="n">variables</span> <span class="o">=</span> <span class="p">{</span><span class="s2">&quot;ensemblId&quot;</span><span class="p">:</span> <span class="n">gene_id</span><span class="p">}</span>
<span class="n">base_url</span> <span class="o">=</span> <span class="s2">&quot;https://api.platform.opentargets.org/api/v4/graphql&quot;</span>
<span class="n">r</span> <span class="o">=</span> <span class="n">requests</span><span class="o">.</span><span class="n">post</span><span class="p">(</span><span class="n">base_url</span><span class="p">,</span> <span class="n">json</span><span class="o">=</span><span class="p">{</span><span class="s2">&quot;query&quot;</span><span class="p">:</span> <span class="n">query_string</span><span class="p">,</span> <span class="s2">&quot;variables&quot;</span><span class="p">:</span> <span class="n">variables</span><span class="p">})</span>
<span class="nb">print</span><span class="p">(</span><span class="n">r</span><span class="o">.</span><span class="n">status_code</span><span class="p">)</span>
<span class="n">api_response</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">loads</span><span class="p">(</span><span class="n">r</span><span class="o">.</span><span class="n">text</span><span class="p">)</span>
<span class="nb">print</span><span class="p">(</span><span class="n">api_response</span><span class="p">)</span>
</code></pre></div>
<p>Lastly we turn to R, which again gets around <code>httr::content(r)</code> for <code>iconvlist()</code> with <code>iconv()</code>.</p>
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">httr</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">jsonlite</span><span class="p">)</span>

<span class="n">gene_id</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;ENSG00000164308&quot;</span>
<span class="n">query_string</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s">&quot;</span>
<span class="s">  query target($ensemblId: String!){</span>
<span class="s">    target(ensemblId: $ensemblId){</span>
<span class="s">      id</span>
<span class="s">      approvedSymbol</span>
<span class="s">      biotype</span>
<span class="s">      geneticConstraint {</span>
<span class="s">        constraintType</span>
<span class="s">        exp</span>
<span class="s">        obs</span>
<span class="s">        score</span>
<span class="s">        oe</span>
<span class="s">        oeLower</span>
<span class="s">        oeUpper</span>
<span class="s">      }</span>
<span class="s">      tractability {</span>
<span class="s">        label</span>
<span class="s">        modality</span>
<span class="s">        value</span>
<span class="s">      }</span>
<span class="s">    }</span>
<span class="s">  }</span>
<span class="s">&quot;</span>
<span class="n">base_url</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;https://api.platform.opentargets.org/api/v4/graphql&quot;</span>
<span class="n">variables</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="s">&quot;ensemblId&quot;</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">gene_id</span><span class="p">)</span>
<span class="n">post_body</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">query</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">query_string</span><span class="p">,</span><span class="w"> </span><span class="n">variables</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">variables</span><span class="p">)</span>
<span class="n">r</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">httr</span><span class="o">::</span><span class="nf">POST</span><span class="p">(</span><span class="n">url</span><span class="o">=</span><span class="n">base_url</span><span class="p">,</span><span class="w"> </span><span class="n">body</span><span class="o">=</span><span class="n">post_body</span><span class="p">,</span><span class="w"> </span><span class="n">encode</span><span class="o">=</span><span class="s">&#39;json&#39;</span><span class="p">)</span>
<span class="n">data</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">iconv</span><span class="p">(</span><span class="n">r</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;ASCII&quot;</span><span class="p">)</span>
<span class="n">content</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">jsonlite</span><span class="o">::</span><span class="nf">fromJSON</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
</code></pre></div>
<h2 id="pgs-searchgpt">PGS (searchGPT)</h2>
<p><a href="https://frontlinegenomics.com/how-to-perform-polygenic-risk-score-analysis/"><img alt="perform polygenic risk score analysis" src="https://images.openai.com/thumbnails/0e8e5ab5caec6614efac03beb20c426c.png" /></a></p>
<p>To calculate a Polygenic Risk Score (PRS) using PLINK, you typically employ a weighted sum approach, where each SNP contributes according to its effect size (e.g., from a GWAS meta-analysis). PLINK facilitates this process through the <code>--score</code> command, which computes the PRS for each individual based on specified SNPs and their associated effect sizes.(<a href="https://choishingwan.github.io/PRS-Tutorial/plink/?utm_source=chatgpt.com" title="PLINK - Basic Tutorial for Polygenic Risk Score Analyses">Sam Choi</a>)</p>
<h3 id="step-by-step-guide-to-calculating-prs-with-plink">Step-by-Step Guide to Calculating PRS with PLINK</h3>
<h4 id="1-prepare-your-input-files">1. Prepare Your Input Files</h4>
<ul>
<li>
<p><strong>Genotype Data</strong>: Ensure your genotype data is in PLINK's binary format (<code>.bed</code>, <code>.bim</code>, <code>.fam</code>).</p>
</li>
<li>
<p><strong>SNP Information File</strong>: This file should include SNP identifiers, effect alleles, and effect sizes (betas or odds ratios) from the base GWAS study. The format typically includes columns for SNP ID, effect allele, non-effect allele, effect size, and standard error.</p>
</li>
</ul>
<h4 id="2-format-the-snp-information-file">2. Format the SNP Information File</h4>
<p>The SNP information file should be structured with columns corresponding to:(<a href="https://choishingwan.github.io/PRS-Tutorial/plink/?utm_source=chatgpt.com" title="PLINK - Basic Tutorial for Polygenic Risk Score Analyses">Sam Choi</a>)</p>
<ol>
<li>
<p><strong>SNP ID</strong>: Unique identifier for each SNP.</p>
</li>
<li>
<p><strong>Effect Allele</strong>: The allele associated with the trait.</p>
</li>
<li>
<p><strong>Non-Effect Allele</strong>: The alternative allele.</p>
</li>
<li>
<p><strong>Effect Size</strong>: The estimated effect size (beta or odds ratio) for the effect allele.</p>
</li>
<li>
<p><strong>Standard Error</strong>: The standard error of the effect size estimate.</p>
</li>
</ol>
<p>Ensure that the SNP identifiers in this file match those in your genotype data.</p>
<h4 id="3-execute-the-prs-calculation">3. Execute the PRS Calculation</h4>
<p>Use the following PLINK command to calculate the PRS:(<a href="https://choishingwan.github.io/PRS-Tutorial/plink/?utm_source=chatgpt.com" title="PLINK - Basic Tutorial for Polygenic Risk Score Analyses">Sam Choi</a>)</p>
<div class="highlight"><pre><span></span><code>plink<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--bfile<span class="w"> </span>your_genotype_data<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--score<span class="w"> </span>snp_info_file<span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="m">6</span><span class="w"> </span>header<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--out<span class="w"> </span>prscore_output
</code></pre></div>
<p>In this command:(<a href="https://choishingwan.github.io/PRS-Tutorial/plink/?utm_source=chatgpt.com" title="PLINK - Basic Tutorial for Polygenic Risk Score Analyses">Sam Choi</a>)</p>
<ul>
<li>
<p><code>--bfile your_genotype_data</code>: Specifies the base name of your binary genotype files (<code>your_genotype_data.bed</code>, <code>.bim</code>, <code>.fam</code>).</p>
</li>
<li>
<p><code>--score snp_info_file 2 4 6 header</code>: Indicates the SNP information file and specifies that the second column contains SNP IDs, the fourth column contains effect alleles, and the sixth column contains effect sizes. The <code>header</code> option tells PLINK to skip the first row.</p>
</li>
<li>
<p><code>--out prscore_output</code>: Designates the prefix for the output files.</p>
</li>
</ul>
<p>PLINK will generate output files with the specified prefix, containing the calculated PRS for each individual.(<a href="https://www.frontiersin.org/journals/genetics/articles/10.3389/fgene.2022.818574/full?utm_source=chatgpt.com" title="Frontiers | Calculating Polygenic Risk Scores (PRS) in UK Biobank: A Practical Guide for Epidemiologists">Frontiers</a>)</p>
<h4 id="4-interpret-the-output">4. Interpret the Output</h4>
<p>The output file will include columns such as:</p>
<ul>
<li>
<p><strong>FID</strong>: Family ID.</p>
</li>
<li>
<p><strong>IID</strong>: Individual ID.</p>
</li>
<li>
<p><strong>SNPs Count</strong>: Number of SNPs contributing to the PRS.(<a href="https://choishingwan.github.io/PRS-Tutorial/plink/?utm_source=chatgpt.com" title="PLINK - Basic Tutorial for Polygenic Risk Score Analyses">Sam Choi</a>)</p>
</li>
<li>
<p><strong>PRS</strong>: The calculated polygenic risk score for each individual.</p>
</li>
</ul>
<p>You can further analyze these scores to assess the genetic risk associated with the trait of interest.</p>
<h4 id="5-optional-apply-p-value-thresholding">5. Optional: Apply P-value Thresholding</h4>
<p>To refine your PRS, you can apply P-value thresholds to include only SNPs with P-values below a certain cutoff. This can be done using the <code>--q-score-range</code> option in PLINK, which allows you to specify a range of P-values for inclusion.(<a href="https://choishingwan.github.io/PRS-Tutorial/plink/?utm_source=chatgpt.com" title="PLINK - Basic Tutorial for Polygenic Risk Score Analyses">Sam Choi</a>)</p>
<p>For example, to include SNPs with P-values less than 0.05:(<a href="https://choishingwan.github.io/PRS-Tutorial/plink/?utm_source=chatgpt.com" title="PLINK - Basic Tutorial for Polygenic Risk Score Analyses">Sam Choi</a>)</p>
<div class="highlight"><pre><span></span><code>plink<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--bfile<span class="w"> </span>your_genotype_data<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--score<span class="w"> </span>snp_info_file<span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="m">6</span><span class="w"> </span>header<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--q-score-range<span class="w"> </span><span class="m">0</span>.05<span class="w"> </span><span class="m">0</span>.05<span class="w"> </span><span class="se">\</span>
<span class="w">  </span>--out<span class="w"> </span>prscore_output_thresholded
</code></pre></div>
<p>This command will calculate the PRS using only SNPs with P-values less than 0.05.(<a href="https://choishingwan.github.io/PRS-Tutorial/plink/?utm_source=chatgpt.com" title="PLINK - Basic Tutorial for Polygenic Risk Score Analyses">Sam Choi</a>)</p>
<h4 id="6-validate-and-interpret-the-prs">6. Validate and Interpret the PRS</h4>
<p>After calculating the PRS, it's important to validate its predictive power and interpret its significance. This can involve:</p>
<ul>
<li>
<p><strong>Association Analysis</strong>: Assessing the association between the PRS and the trait or disease phenotype.</p>
</li>
<li>
<p><strong>Model Evaluation</strong>: Evaluating the performance of the PRS in predicting outcomes using metrics like R or area under the ROC curve.</p>
</li>
<li>
<p><strong>Stratification</strong>: Dividing the population into risk groups based on PRS percentiles to examine differential outcomes.</p>
</li>
</ul>
<p>For more detailed guidance on these steps, refer to the tutorial provided by Choi et al. .(<a href="https://choishingwan.github.io/PRS-Tutorial/?utm_source=chatgpt.com" title="Basic Tutorial for Polygenic Risk Score Analyses">Sam Choi</a>)</p>
<p>If you're working with imputed data, consider using PLINK 2.0, which offers enhanced support for imputed genotypes and additional functionalities for PRS analysis. For comprehensive instructions on using PLINK 2.0 for PRS calculation, refer to the tutorial by Choi et al. .(<a href="https://www.frontiersin.org/journals/genetics/articles/10.3389/fgene.2022.818574/full?utm_source=chatgpt.com" title="Frontiers | Calculating Polygenic Risk Scores (PRS) in UK Biobank: A Practical Guide for Epidemiologists">Frontiers</a>)</p>
<p>For a practical example of calculating a PRS using UK Biobank data, refer to the guide by Jennifer Collister and Xiaonan Liu .(<a href="https://2cjenn.github.io/PRS_Pipeline/?utm_source=chatgpt.com" title="PRS Pipeline">2cjenn.github.io</a>)</p>
<p>If you need assistance with specific aspects of PRS calculation, such as quality control, visualization, or interpretation, feel free to ask!</p>
<h2 id="rentrez">rentrez</h2>
<p>The relevant URLs are as follows, </p>
<ul>
<li><a href="https://cran.r-project.org/web/packages/rentrez/vignettes/rentrez_tutorial.html">https://cran.r-project.org/web/packages/rentrez/vignettes/rentrez_tutorial.html</a></li>
<li><a href="https://pubmed.ncbi.nlm.nih.gov/">https://pubmed.ncbi.nlm.nih.gov/</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/pmc/pmctopmid/">https://www.ncbi.nlm.nih.gov/pmc/pmctopmid/</a></li>
</ul>
<p>with example code,
<div class="highlight"><pre><span></span><code><span class="nf">library</span><span class="p">(</span><span class="n">rentrez</span><span class="p">)</span>
<span class="nf">entrez_dbs</span><span class="p">()</span>
<span class="nf">entrez_db_links</span><span class="p">(</span><span class="s">&quot;pubmed&quot;</span><span class="p">)</span>
<span class="n">pubmed_fields</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">entrez_db_searchable</span><span class="p">(</span><span class="s">&quot;pubmed&quot;</span><span class="p">)</span>
<span class="c1"># set_entrez_key(&quot;&quot;)</span>
<span class="nf">Sys.getenv</span><span class="p">(</span><span class="s">&quot;ENTREZ_KEY&quot;</span><span class="p">)</span>
<span class="n">term</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;pQTLs OR (protein AND quantitative AND trait AND loci) AND human [MH] AND (plasma OR Serum)&quot;</span>
<span class="n">r</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">entrez_search</span><span class="p">(</span><span class="n">db</span><span class="o">=</span><span class="s">&quot;pubmed&quot;</span><span class="p">,</span><span class="n">term</span><span class="o">=</span><span class="n">term</span><span class="p">,</span><span class="n">use_history</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="nf">class</span><span class="p">(</span><span class="n">r</span><span class="p">)</span>
<span class="nf">names</span><span class="p">(</span><span class="n">r</span><span class="p">)</span>
<span class="nf">with</span><span class="p">(</span><span class="n">r</span><span class="p">,</span><span class="n">web_history</span><span class="p">)</span>
<span class="nf">unlink</span><span class="p">(</span><span class="nf">paste</span><span class="p">(</span><span class="s">&quot;pubmed&quot;</span><span class="p">,</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;fetch&quot;</span><span class="p">,</span><span class="s">&quot;summary&quot;</span><span class="p">),</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;.&quot;</span><span class="p">))</span>
<span class="n">fields</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;uid&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;pubdate&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;sortfirstauthor&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;title&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;source&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;volume&quot;</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;pages&quot;</span><span class="p">)</span>
<span class="nf">for</span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="nf">seq</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="nf">with</span><span class="p">(</span><span class="n">r</span><span class="p">,</span><span class="n">count</span><span class="p">),</span><span class="m">50</span><span class="p">))</span>
<span class="p">{</span>
<span class="w">  </span><span class="nf">cat</span><span class="p">(</span><span class="n">i</span><span class="m">+49</span><span class="p">,</span><span class="w"> </span><span class="s">&quot;records downloaded\r&quot;</span><span class="p">)</span>
<span class="w">  </span><span class="n">f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">entrez_fetch</span><span class="p">(</span><span class="n">db</span><span class="o">=</span><span class="s">&quot;pubmed&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">web_history</span><span class="o">=</span><span class="nf">with</span><span class="p">(</span><span class="n">r</span><span class="p">,</span><span class="n">web_history</span><span class="p">),</span><span class="w"> </span><span class="n">rettype</span><span class="o">=</span><span class="s">&quot;text&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">retmax</span><span class="o">=</span><span class="m">50</span><span class="p">,</span><span class="w"> </span><span class="n">retstart</span><span class="o">=</span><span class="n">i</span><span class="p">)</span>
<span class="w">  </span><span class="nf">write.table</span><span class="p">(</span><span class="n">f</span><span class="p">,</span><span class="w"> </span><span class="n">col.names</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="o">=</span><span class="s">&quot;pubmed.fetch&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">append</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="w">  </span><span class="n">s</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">entrez_summary</span><span class="p">(</span><span class="n">db</span><span class="o">=</span><span class="s">&quot;pubmed&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">web_history</span><span class="o">=</span><span class="nf">with</span><span class="p">(</span><span class="n">r</span><span class="p">,</span><span class="n">web_history</span><span class="p">),</span><span class="w"> </span><span class="n">rettype</span><span class="o">=</span><span class="s">&quot;text&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">retmax</span><span class="o">=</span><span class="m">50</span><span class="p">,</span><span class="w"> </span><span class="n">retstart</span><span class="o">=</span><span class="n">i</span><span class="p">)</span>
<span class="w">  </span><span class="n">e</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">extract_from_esummary</span><span class="p">(</span><span class="n">s</span><span class="p">,</span><span class="w"> </span><span class="n">fields</span><span class="p">)</span>
<span class="w">  </span><span class="nf">write.table</span><span class="p">(</span><span class="nf">t</span><span class="p">(</span><span class="n">e</span><span class="p">),</span><span class="w"> </span><span class="n">col.names</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">row.names</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="o">=</span><span class="s">&quot;pubmed.summary&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">append</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">)</span>
<span class="p">}</span>
<span class="n">id</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">600807</span>
<span class="n">upload</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">entrez_post</span><span class="p">(</span><span class="n">db</span><span class="o">=</span><span class="s">&quot;omim&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">id</span><span class="o">=</span><span class="n">id</span><span class="p">)</span>
<span class="n">asthma_variants</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">entrez_link</span><span class="p">(</span><span class="n">dbfrom</span><span class="o">=</span><span class="s">&quot;omim&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">db</span><span class="o">=</span><span class="s">&quot;clinvar&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">cmd</span><span class="o">=</span><span class="s">&quot;neighbor_history&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">web_history</span><span class="o">=</span><span class="n">upload</span><span class="p">)</span>
<span class="n">asthma_variants</span>
<span class="n">snp_links</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">entrez_link</span><span class="p">(</span><span class="n">dbfrom</span><span class="o">=</span><span class="s">&quot;clinvar&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">db</span><span class="o">=</span><span class="s">&quot;snp&quot;</span><span class="p">,</span><span class="w"> </span><span class="n">web_history</span><span class="o">=</span><span class="n">asthma_variants</span><span class="o">$</span><span class="n">web_histories</span><span class="o">$</span><span class="n">omim_clinvar</span><span class="p">,</span><span class="w"> </span><span class="n">cmd</span><span class="o">=</span><span class="s">&quot;neighbor_history&quot;</span><span class="p">)</span>
<span class="n">all_links</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">entrez_link</span><span class="p">(</span><span class="n">dbfrom</span><span class="o">=</span><span class="s">&#39;pubmed&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">id</span><span class="o">=</span><span class="n">id</span><span class="p">,</span><span class="w"> </span><span class="n">db</span><span class="o">=</span><span class="s">&#39;all&#39;</span><span class="p">)</span>
</code></pre></div></p>
<h2 id="roadmap">Roadmap</h2>
<p><a href="http://www.roadmapepigenomics.org/">http://www.roadmapepigenomics.org/</a></p>
<h2 id="snakemake-workflow-catalogue">snakemake workflow catalogue</h2>
<p><a href="https://snakemake.github.io/snakemake-workflow-catalog/">https://snakemake.github.io/snakemake-workflow-catalog/</a></p>
<h2 id="twas">TWAS</h2>
<ul>
<li>MetaXcan, <a href="https://github.com/hakyimlab/MetaXcan">https://github.com/hakyimlab/MetaXcan</a></li>
<li>FUSION, <a href="http://gusevlab.org/projects/fusion/">http://gusevlab.org/projects/fusion/</a></li>
<li>OmicsPred, <a href="https://www.omicspred.org/">https://www.omicspred.org/</a></li>
<li>PredictDB data repository, <a href="http://predictdb.org/">http://predictdb.org/</a></li>
<li>TWAS-hub, <a href="http://twas-hub.org/">http://twas-hub.org/</a></li>
</ul>
<h2 id="uk10k-imputation">UK10K imputation</h2>
<p>Sanger Imputation Service, <a href="https://imputation.sanger.ac.uk/">https://imputation.sanger.ac.uk/</a></p>
<h2 id="other-links">Other links</h2>
<ul>
<li><a href="https://www.ucl.ac.uk/health-informatics/caliber">CALIBER</a>.</li>
<li>deCode <a href="https://www.decode.com/summarydata/">summary statistics</a></li>
<li><a href="https://usegalaxy.eu/">Galaxy Europe</a></li>
<li><a href="https://ftp-trace.ncbi.nlm.nih.gov/ReferenceSamples/giab/data/">Genome in a Bottle WGS samples</a></li>
<li><a href="https://portal.genego.com/">genego</a>.</li>
<li>MVP GWAS summary statistics, <a href="https://ftp.ncbi.nlm.nih.gov/dbgap/studies/phs002453/analyses/">https://ftp.ncbi.nlm.nih.gov/dbgap/studies/phs002453/analyses/</a>.</li>
<li><a href="https://github.com/aehrc">The Australian e-Health Research Centre</a>.</li>
<li><a href="https://github.com/hmgu-itg">Institute of Translational Genomics</a> and <a href="https://omicscience.org/">omicscience</a>.</li>
<li><a href="http://bioinformatics.sdstate.edu/idep/">idep</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/myncbi/">NCBI account</a> (<a href="https://www.ncbi.nlm.nih.gov/account/settings/">settings</a>).</li>
</ul></div>
            </div>
        </div>

        <footer class="col-md-12">
            <hr>
            <p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>
        <script src="../js/jquery-3.6.0.min.js"></script>
        <script src="../js/bootstrap.min.js"></script>
        <script>
            var base_url = "..",
                shortcuts = {"help": 191, "next": 78, "previous": 80, "search": 83};
        </script>
        <script src="../js/base.js"></script>
        <script src="../javascripts/config.js"></script>
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
        <script src="../search/main.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="searchModalLabel" aria-hidden="true">
    <div class="modal-dialog modal-lg">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="searchModalLabel">Search</h4>
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            </div>
            <div class="modal-body">
                <p>From here you can search these documents. Enter your search terms below.</p>
                <form>
                    <div class="form-group">
                        <input type="search" class="form-control" placeholder="Search..." id="mkdocs-search-query" title="Type search term here">
                    </div>
                </form>
                <div id="mkdocs-search-results" data-no-results-text="No results found"></div>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div><div class="modal" id="mkdocs_keyboard_modal" tabindex="-1" role="dialog" aria-labelledby="keyboardModalLabel" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="keyboardModalLabel">Keyboard Shortcuts</h4>
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            </div>
            <div class="modal-body">
              <table class="table">
                <thead>
                  <tr>
                    <th style="width: 20%;">Keys</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td class="help shortcut"><kbd>?</kbd></td>
                    <td>Open this help</td>
                  </tr>
                  <tr>
                    <td class="next shortcut"><kbd>n</kbd></td>
                    <td>Next page</td>
                  </tr>
                  <tr>
                    <td class="prev shortcut"><kbd>p</kbd></td>
                    <td>Previous page</td>
                  </tr>
                  <tr>
                    <td class="search shortcut"><kbd>s</kbd></td>
                    <td>Search</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div>

    </body>
</html>
