<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="../img/favicon.ico">
        <title>CAD/MI - Omics-analysis</title>
        <link href="../css/bootstrap.min.css" rel="stylesheet">
        <link href="../css/font-awesome.min.css" rel="stylesheet">
        <link href="../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css">
        <link href="../stylesheets/extra.css" rel="stylesheet">
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/Bash.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/HTML.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/Java.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/JavaScript.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/Perl.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/Python.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/R.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/rust.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/yaml.min.js"></script>
        <script>hljs.highlightAll();</script> 
    </head>

    <body>
        <div class="navbar fixed-top navbar-expand-lg navbar-dark bg-dark">
            <div class="container">
                <a class="navbar-brand" href="..">Omics-analysis</a>
                <!-- Expander button -->
                <button type="button" class="navbar-toggler" data-toggle="collapse" data-target="#navbar-collapse">
                    <span class="navbar-toggler-icon"></span>
                </button>

                <!-- Expanded navigation -->
                <div id="navbar-collapse" class="navbar-collapse collapse">
                        <!-- Main navigation -->
                        <ul class="nav navbar-nav">
                            <li class="navitem">
                                <a href=".." class="nav-link"></a>
                            </li>
                            <li class="navitem">
                                <a href="../resources/" class="nav-link">Resources</a>
                            </li>
                            <li class="dropdown active">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">Cases <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li>
    <a href="../AF/" class="dropdown-item">Atrial Fibrillation</a>
</li>
                                    
<li>
    <a href="../BMI/" class="dropdown-item">BMI</a>
</li>
                                    
<li>
    <a href="./" class="dropdown-item active">CAD/MI</a>
</li>
                                    
<li>
    <a href="../T2D/" class="dropdown-item">T2D</a>
</li>
                                </ul>
                            </li>
                            <li class="navitem">
                                <a href="../genomics/" class="nav-link">Genomics</a>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown">Proteomics <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li>
    <a href="../proteomics/" class="dropdown-item">Proteomics</a>
</li>
                                    
<li>
    <a href="../R-Programming-for-Mass-Spectrometry/" class="dropdown-item">R Programming for Mass Spectrometry</a>
</li>
                                </ul>
                            </li>
                        </ul>

                    <ul class="nav navbar-nav ml-auto">
                        <li class="nav-item">
                            <a href="#" class="nav-link" data-toggle="modal" data-target="#mkdocs_search_modal">
                                <i class="fa fa-search"></i> Search
                            </a>
                        </li>
                            <li class="nav-item">
                                <a rel="prev" href="../BMI/" class="nav-link">
                                    <i class="fa fa-arrow-left"></i> Previous
                                </a>
                            </li>
                            <li class="nav-item">
                                <a rel="next" href="../T2D/" class="nav-link">
                                    Next <i class="fa fa-arrow-right"></i>
                                </a>
                            </li>
                            <li class="nav-item">
                                <a href="https://github.com/jinghuazhao/Omics-analysis" class="nav-link"><i class="fa fa-github"></i> GitHub</a>
                            </li>
                    </ul>
                </div>
            </div>
        </div>

        <div class="container">
            <div class="row">
                    <div class="col-md-3"><div class="navbar-light navbar-expand-md bs-sidebar hidden-print affix" role="complementary">
    <div class="navbar-header">
        <button type="button" class="navbar-toggler collapsed" data-toggle="collapse" data-target="#toc-collapse" title="Table of Contents">
            <span class="fa fa-angle-down"></span>
        </button>
    </div>

    
    <div id="toc-collapse" class="navbar-collapse collapse card bg-secondary">
        <ul class="nav flex-column">
            
            <li class="nav-item" data-level="1"><a href="#cadmi" class="nav-link">CAD/MI</a>
              <ul class="nav flex-column">
            <li class="nav-item" data-level="2"><a href="#mr" class="nav-link">MR</a>
              <ul class="nav flex-column">
              </ul>
            </li>
            <li class="nav-item" data-level="2"><a href="#pathway-analysis" class="nav-link">Pathway analysis</a>
              <ul class="nav flex-column">
              </ul>
            </li>
              </ul>
            </li>
        </ul>
    </div>
</div></div>
                    <div class="col-md-9" role="main">

<h1 id="cadmi">CAD/MI</h1>
<p>The summary statistics is from http://www.cardiogramplusc4d.org/, notably
* <a href="http://www.cardiogramplusc4d.org/media/cardiogramplusc4d-consortium/data-downloads/cardiogram_gwas_results.zip">CARDIoGRAM GWA meta-analysis</a></p>
<blockquote>
<p>Schunkert H, König IR, Kathiresan S, Reilly MP, Assimes TL, Holm H et al. Large-scale association analysis identifies 13 new susceptibility loci for coronary artery disease. Nat Genet. 2011 43: 333-338</p>
</blockquote>
<ul>
<li><a href="http://www.cardiogramplusc4d.org/media/cardiogramplusc4d-consortium/data-downloads/cad.additive.Oct2015.pub.zip">CARDIoGRAMplusC4D 1000 Genomes-based GWAS</a></li>
</ul>
<blockquote>
<p>Nikpey M, Goel A, Won H, Hall LM, Willenborg C, Kanoni S, Saleheen S, et al. A comprehensive 1000 Genomes–based genome-wide association meta-analysis of coronary artery disease. Nat Genet 2015 47:1121-1130</p>
</blockquote>
<ul>
<li><a href="http://www.cardiogramplusc4d.org/media/cardiogramplusc4d-consortium/data-downloads/UKBB.GWAS1KG.EXOME.CAD.SOFT.META.PublicRelease.300517.txt.gz">UKBB.GWAS1KG.EXOME.CAD.SOFT.META.PublicRelease.300517</a></li>
</ul>
<blockquote>
<p>Nelson CP, Goel A, Butterworth AS, Kanoni S, Webb TR, et al. Association analyses based on false discovery rate implicate new loci for coronary artery disease. Nat Genet 2017 Jul 17 49(9): 1385-1391. doi: 10.1038/ng.3913</p>
</blockquote>
<p>For the CARDIoGRAMplusC4D 1000 Genomes-based GWAS, the setup is
<div class="highlight"><pre><span></span><code>wget<span class="w"> </span>http://www.cardiogramplusc4d.org/media/cardiogramplusc4d-consortium/data-downloads/cad.additive.Oct2015.pub.zip
unzip<span class="w"> </span>cad.additive.Oct2015.pub.zip
</code></pre></div>
giving `cad.add.160614.website.txt.</p>
<h2 id="mr">MR</h2>
<p>We examine MMP-12 and CAD as in the Nature paper, the input data is generated with <a href="MMP12.sh">MMP12.sh</a>.</p>
<p>THe analogous coding for TwoSampleMR as with MendelianRandomization in <a href="https://github.com/jinghuazhao/software-notes">software-notes</a>
is contained in <a href="MMP12.R">MMP12.R</a>.</p>
<p>We could also follow Generalised Summary-data-based Mendelian Randomisation (<a href="http://cnsgenomics.com/software/gcta/#Mendelianrandomisation">GSMR</a>).
<div class="highlight"><pre><span></span><code><span class="c1"># 16-7-2019 JHZ</span>

awk<span class="w"> </span><span class="s1">&#39;</span>
<span class="s1">{</span>
<span class="s1">   if (NR==1) print &quot;SNP&quot;, &quot;A1&quot;, &quot;A2&quot;, &quot;freq&quot;, &quot;b&quot;, &quot;se&quot;, &quot;p&quot;, &quot;N&quot;;</span>
<span class="s1">   else {</span>
<span class="s1">     CHR=$2</span>
<span class="s1">     POS=$3</span>
<span class="s1">     a1=$4</span>
<span class="s1">     a2=$5</span>
<span class="s1">     if (a1&gt;a2) snpid=&quot;chr&quot; CHR &quot;:&quot; POS &quot;_&quot; a2 &quot;_&quot; a1;</span>
<span class="s1">     else snpid=&quot;chr&quot; CHR &quot;:&quot; POS &quot;_&quot; a1 &quot;_&quot; a2</span>
<span class="s1">     $1=snpid</span>
<span class="s1">     print snpid, a1, a2, $6, $9, $10, $11, 185000</span>
<span class="s1">   }</span>
<span class="s1">}&#39;</span><span class="w"> </span>CAD/cad.add.160614.website.txt<span class="w"> </span>&gt;<span class="w"> </span>gsmr_outcome.txt

<span class="c1"># awk &#39;NR==1&#39; CAD/cad.add.160614.website.txt | sed &#39;s|\t|\n|g&#39; | awk &#39;{print &quot;# &quot; NR, $1}&#39;</span>
<span class="c1"># 1 markername</span>
<span class="c1"># 2 chr</span>
<span class="c1"># 3 bp_hg19</span>
<span class="c1"># 4 effect_allele</span>
<span class="c1"># 5 noneffect_allele</span>
<span class="c1"># 6 effect_allele_freq</span>
<span class="c1"># 7 median_info</span>
<span class="c1"># 8 model</span>
<span class="c1"># 9 beta</span>
<span class="c1"># 10 se_dgc</span>
<span class="c1"># 11 p_dgc</span>
<span class="c1"># 12 het_pvalue</span>
<span class="c1"># 13 n_studies</span>

xport<span class="w"> </span><span class="nv">INF</span><span class="o">=</span>/rds/project/jmmh2/rds-jmmh2-projects/olink_proteomics/scallop/INF
<span class="nb">echo</span><span class="w"> </span><span class="nv">$INF</span>/INTERVAL/INTERVAL<span class="w"> </span>&gt;<span class="w"> </span>gsmr_ref_data
<span class="nb">echo</span><span class="w"> </span>IL.6<span class="w"> </span><span class="nv">$INF</span>/work/IL.6.ma<span class="w"> </span>&gt;<span class="w"> </span>gsmr_exposure
<span class="nb">echo</span><span class="w"> </span>CAD<span class="w"> </span>gsmr_outcome.txt<span class="w"> </span>&gt;<span class="w"> </span>gsmr_outcome

gcta-1.9<span class="w"> </span>--mbfile<span class="w"> </span>gsmr_ref_data<span class="w"> </span>--gsmr-file<span class="w"> </span>gsmr_exposure<span class="w"> </span>gsmr_outcome<span class="w"> </span>--gsmr-direction<span class="w"> </span><span class="m">0</span><span class="w"> </span>--effect-plot<span class="w"> </span>--out<span class="w"> </span>gsmr_result

R<span class="w"> </span>--no-save<span class="w"> </span>-q<span class="w"> </span><span class="s">&lt;&lt;END</span>
<span class="s">  source(&quot;http://cnsgenomics.com/software/gcta/static/gsmr_plot.r&quot;)</span>
<span class="s">  gsmr_data &lt;- read_gsmr_data(&quot;gsmr_result.eff_plot.gz&quot;)</span>
<span class="s">  gsmr_summary(gsmr_data)</span>
<span class="s">  plot_gsmr_effect(gsmr_data, &quot;IL.6&quot;, &quot;cad&quot;, colors()[75])</span>
<span class="s">END</span>
</code></pre></div>
where we use IL.6 as exposure data; the effect-size plots are generated.</p>
<p>Note that <code>gcta-1.9</code> indcates GCTA 1.9.x is necessary since the <code>--mfile</code> option is not recognised by GCTA 1.26.0.</p>
<h2 id="pathway-analysis">Pathway analysis</h2>
<p>MAGMA is illustrated here,</p>
<p>The GWAS summary data can either be formatted with R,
<div class="highlight"><pre><span></span><code><span class="n">R</span><span class="w"> </span><span class="o">--</span><span class="n">no</span><span class="o">-</span><span class="n">save</span><span class="w"> </span><span class="o">&lt;&lt;</span><span class="n">END</span>
<span class="n">d</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">read.delim</span><span class="p">(</span><span class="s">&quot;`cad.add.160614.website.txt&quot;</span><span class="p">,</span><span class="n">as.is</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="n">db</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s">&quot;CAD&quot;</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">d</span><span class="p">[</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;SNP&quot;</span><span class="p">,</span><span class="s">&quot;CHR&quot;</span><span class="p">,</span><span class="s">&quot;BP&quot;</span><span class="p">)],</span><span class="n">file</span><span class="o">=</span><span class="nf">paste0</span><span class="p">(</span><span class="n">db</span><span class="p">,</span><span class="s">&quot;.snploc&quot;</span><span class="p">),</span><span class="n">quote</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="n">row.name</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="n">col.names</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">)</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">d</span><span class="p">[</span><span class="nf">c</span><span class="p">(</span><span class="s">&quot;SNP&quot;</span><span class="p">,</span><span class="s">&quot;P&quot;</span><span class="p">,</span><span class="s">&quot;NOBS&quot;</span><span class="p">)],</span><span class="n">file</span><span class="o">=</span><span class="nf">paste0</span><span class="p">(</span><span class="n">db</span><span class="p">,</span><span class="s">&quot;.pval&quot;</span><span class="p">),</span><span class="n">quote</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="n">row.name</span><span class="o">=</span><span class="kc">FALSE</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">)</span>
<span class="n">END</span>
</code></pre></div>
or more efficiently with bash before pathway analysis
<div class="highlight"><pre><span></span><code>awk<span class="w"> </span>-vOFS<span class="o">=</span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;{print $2,$1,$4}&#39;</span><span class="w"> </span>g1000_eur.bim<span class="w"> </span>&gt;<span class="w"> </span>g1000_eur.snploc
awk<span class="w"> </span>-vOFS<span class="o">=</span><span class="s2">&quot;\t&quot;</span><span class="w"> </span><span class="s1">&#39;{if(NR==1) print &quot;SNP&quot;, &quot;P&quot;, &quot;NOBS&quot;; else print $1,$11,1000}&#39;</span><span class="w"> </span><span class="sb">`</span>cad.add.160614.website.txt<span class="w"> </span>&gt;<span class="w"> </span>CAD.pval

<span class="c1"># Annotation</span>
magma<span class="w"> </span>--annotate<span class="w"> </span><span class="nv">window</span><span class="o">=</span><span class="m">50</span>,50<span class="w"> </span>--snp-loc<span class="w"> </span>g1000_eur.snploc<span class="w"> </span>--gene-loc<span class="w"> </span>NCBI37.3.gene.loc<span class="w"> </span>--out<span class="w"> </span>CAD
<span class="c1"># Gene analysis - SNP p-values</span>
magma<span class="w"> </span>--bfile<span class="w"> </span>g1000_eur<span class="w"> </span>--pval<span class="w"> </span>CAD.pval<span class="w"> </span><span class="nv">ncol</span><span class="o">=</span>NOBS<span class="w"> </span>--gene-annot<span class="w"> </span>CAD.genes.annot<span class="w"> </span>--out<span class="w"> </span>CAD
<span class="c1"># Pathway analysis</span>
<span class="c1"># http://software.broadinstitute.org/gsea/downloads.jsp</span>
magma<span class="w"> </span>--gene-results<span class="w"> </span>CAD.genes.raw<span class="w"> </span>--set-annot<span class="w"> </span>msigdb.v6.2.entrez.gmt<span class="w"> </span>self-contained<span class="w"> </span>--model<span class="w"> </span>fwer<span class="w"> </span>--out<span class="w"> </span>CAD
</code></pre></div></p></div>
            </div>
        </div>

        <footer class="col-md-12">
            <hr>
            <p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>
        <script src="../js/jquery-3.6.0.min.js"></script>
        <script src="../js/bootstrap.min.js"></script>
        <script>
            var base_url = "..",
                shortcuts = {"help": 191, "next": 78, "previous": 80, "search": 83};
        </script>
        <script src="../js/base.js"></script>
        <script src="../javascripts/config.js"></script>
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
        <script src="../search/main.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="searchModalLabel" aria-hidden="true">
    <div class="modal-dialog modal-lg">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="searchModalLabel">Search</h4>
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            </div>
            <div class="modal-body">
                <p>From here you can search these documents. Enter your search terms below.</p>
                <form>
                    <div class="form-group">
                        <input type="search" class="form-control" placeholder="Search..." id="mkdocs-search-query" title="Type search term here">
                    </div>
                </form>
                <div id="mkdocs-search-results" data-no-results-text="No results found"></div>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div><div class="modal" id="mkdocs_keyboard_modal" tabindex="-1" role="dialog" aria-labelledby="keyboardModalLabel" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <h4 class="modal-title" id="keyboardModalLabel">Keyboard Shortcuts</h4>
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
            </div>
            <div class="modal-body">
              <table class="table">
                <thead>
                  <tr>
                    <th style="width: 20%;">Keys</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td class="help shortcut"><kbd>?</kbd></td>
                    <td>Open this help</td>
                  </tr>
                  <tr>
                    <td class="next shortcut"><kbd>n</kbd></td>
                    <td>Next page</td>
                  </tr>
                  <tr>
                    <td class="prev shortcut"><kbd>p</kbd></td>
                    <td>Previous page</td>
                  </tr>
                  <tr>
                    <td class="search shortcut"><kbd>s</kbd></td>
                    <td>Search</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div>

    </body>
</html>
